---
title: A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers
keywords:
- genome sequencing
- ddrad sequencing
- nanopore
- Labeo rohita
lang: en-US
date-meta: '2022-02-15'
author-meta:
- Mark A. Arick II
- Chuan-Yu Hsu
- Zenaida Magbanua
- Corrinne E. Grover
- Emma R. Miller
- Olga Pechanova
- Adam Thrash
- Ramey C. Youngblood
- Lauren Ezzell
- Samsul Alam
- John Benzie
- Matthew Hamilton
- Attila Karsi
- Mark L. Lawrence
- Daniel G. Peterson
header-includes: |-
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta name="citation_title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta property="og:title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta property="twitter:title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta name="dc.date" content="2022-02-15" />
  <meta name="citation_publication_date" content="2022-02-15" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Mark A. Arick II" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="0000-0002-7207-3052" />
  <meta name="citation_author" content="Chuan-Yu Hsu" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Zenaida Magbanua" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Corrinne E. Grover" />
  <meta name="citation_author_institution" content="Ecology, Evolution, and Organismal Biology Dept., Iowa State University, Ames, IA, 50010 USA" />
  <meta name="citation_author_orcid" content="0000-0003-3878-5459" />
  <meta name="citation_author" content="Emma R. Miller" />
  <meta name="citation_author_institution" content="Ecology, Evolution, and Organismal Biology Dept., Iowa State University, Ames, IA, 50010 USA" />
  <meta name="citation_author_orcid" content="0000-0001-9009-5303" />
  <meta name="citation_author" content="Olga Pechanova" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Adam Thrash" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="0000-0002-4097-7663" />
  <meta name="citation_author" content="Ramey C. Youngblood" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Lauren Ezzell" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Samsul Alam" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="John Benzie" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Matthew Hamilton" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Attila Karsi" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Mark L. Lawrence" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Daniel G. Peterson" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="0000-0002-0274-5968" />
  <link rel="canonical" href="https://IGBB.github.io/rohu-genome/" />
  <meta property="og:url" content="https://IGBB.github.io/rohu-genome/" />
  <meta property="twitter:url" content="https://IGBB.github.io/rohu-genome/" />
  <meta name="citation_fulltext_html_url" content="https://IGBB.github.io/rohu-genome/" />
  <meta name="citation_pdf_url" content="https://IGBB.github.io/rohu-genome/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://IGBB.github.io/rohu-genome/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/" />
  <meta name="manubot_html_url_versioned" content="https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/" />
  <meta name="manubot_pdf_url_versioned" content="https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
bibliography:
- content/manual-references.json
manubot-output-bibliography: output/references.json
manubot-output-citekeys: output/citations.tsv
manubot-requests-cache-path: ci/cache/requests-cache
manubot-clear-requests-cache: false
...






<small><em>
This manuscript
([permalink](https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/))
was automatically generated
from [IGBB/rohu-genome@5350e6f](https://github.com/IGBB/rohu-genome/tree/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72)
on February 15, 2022.
</em></small>

## Authors



 

Mark A. Arick II<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/0000-0002-7207-3052)
</sup>,
Chuan-Yu Hsu<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Zenaida Magbanua<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Corrinne E. Grover<sup>2
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/0000-0003-3878-5459)
</sup>,
Emma R. Miller<sup>2
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/0000-0001-9009-5303)
</sup>,
Olga Pechanova<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Adam Thrash<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/0000-0002-4097-7663)
</sup>,
Ramey C. Youngblood<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Lauren Ezzell<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Samsul Alam<sup>3
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
John Benzie<sup>3
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Matthew Hamilton<sup>3
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Attila Karsi<sup>3
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Mark L. Lawrence<sup>3
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
</sup>,
Daniel G. Peterson<sup>1
[![ORCID icon](images/orcid.svg){.inline_icon}](https://orcid.org/0000-0002-0274-5968)
</sup>

<small>

</small>


## Abstract 

<!--  Abstract: ~250 words. Concise summary of the results. -->


## Introduction 

<!--  Introduction: no more than 500 words. The introduction should discuss the rationale for the WGS of the particular organism/strain. -->


## Methods & Materials 

<!--  
Methods & Materials: should include the following (note: author may write 
additional subheadings as appropriate â€“ the below points are not designed 
to be a list of complete subheadings, but rather, topics to be covered).
 -  Organism/Strain origin and derivation 
 -  Sequencing methods and preparation details
 -  Data processing methods 
 -  Data availability listing where to access all raw reads, processed data, 
    BWTs, and other relevant data. The assembly should be uploaded to NCBI or 
    ENA.
 -->

### Genome Sequencing & Flowcytometry

ALAM â€“ Rohu blood collection 

#### Flow cytometry 

The genome size of *Labeo rohita* was estimated for five samples via flow
cytometry using trout erythrocyte nuclei (TENs;
https://www.biosure.com/tens.html) as a standard (genome size=6.5pg). For each
sample, nuclei were stabilized in 200 ul of LB01-propidium iodide (PI) buffer as
per [@pmid:24415480], and two drops of TENs standard were used per 50ul of fish
blood. Each sample was measured twice, totaling 10 runs overall. Only
measurements with greater than 5,000 nuclei and a coefficient of variation (CV)
of less than 3% were retained [@pmid:24415480].

#### Illumina short-read sequencing

A total of 2 Âµg of extracted genomic DNA was used for DNA-Seq library
preparation using Illumina TruSeq DNA PCR-free Library Prep Kit (Illumina, San
Diego, CA, USA). The final DNA-Seq library with the insert size range of 350 bp
to 450 bp was submitted to Novogene company (www.en.novogene.com) for total of 2
lanes of PE150 Illumina HiSeq X-Ten (Illumina, San Diego, CA, USA) sequencing.

#### Oxford Nanopore sequencing

For each Nanopore sequencing run, 2 to 2.5 Âµg of genomic DNA was used in the
library preparation with Nanopore Genomic DNA Ligation Sequencing Kit SQK-LSK
109 (Oxford Nanopore Technologies, Oxford, UK). The final library (about 700 to
750 ng) was loaded on a Nanopore Flow Cell R9.4.1 (Oxford Nanopore Technologies,
Oxford, UK) and sequenced on GridION sequencer (Oxford Nanopore Technologies,
Oxford, UK) for 48 hours. A total of 10 flow cell runs were conducted for the
genome assembly.

#### Hi-C sequencing

One hundred Âµl of fish blood was subjected to the Hi-C library preparation using
the Proximo Hi-C Animal Kit (Phase Genomics, Seattle, WA, USA). The final
DNA-Seq library was submitted to Novogene (www.en.novogene.com) for 1 lane of
PE150 Illumina HiSeq X-Ten (Illumina, San Diego, CA, USA) sequencing.

#### Bionano 

### Assembly & Annotation 

Jellyfish (v2.2.10) [@doi:10.1093/bioinformatics/btr011] and GenomeScope (v1.0)
[@pmcid:PMC5870704] estimated the genome size using the Illumina paired reads
digested into 50-mers.

Nanopore data was filtered to remove the control lambda-phage and sequences
shorter than 1000 bases using the nanopack tool suite (v1.0.1)
[@doi:10.1093/bioinformatics/bty149]. Trimmomatic (v0.32)
[@doi:10.1093/bioinformatics/btu170] removed adapters, trimmed low quality
bases, and filtered reads shorter than 85bp. The filtered nanopore data was
assembled into contigs using wtdbg2 (v2.4) [@doi:10.1038/s41592-019-0669-3]. The
contigs were polished using the two iterations of racon (v1.4.0)
[@doi:10.1101/gr.214270.116] with minimap2 (v2.17)
(@doi:10.1093/bioinformatics/bty191) mapping the nanopore reads. The contigs
were further polished using pilon (v1.23) [@doi:10.1371/journal.pone.0112963]
with bwa (v0.7.10) [@doi:10.1093/bioinformatics/btp324] mapping the Illumina
paired reads. The contigs were scaffolded using Bionano Solve (
Solve3.4.1_09262019)
[@url:https://bionanogenomics.com/support-page/bionano-access-software/) and
SALSA (v2.3) [@doi:10.1371/journal.pcbi.1007273]. That scaffolds larger than
10Mb were linked and oriented based on the Onychostoma macrolepis genome
[@doi:10.1111/1755-0998.13190], the most similar chromosome assembly available
on NCBI, using ragtag (v1.1.1) [@doi:10.1186/s13059-019-1829-6].

RepeatModeler (v2.0.1) [@doi:10.1073/pnas.1921046117] and RepeatMasker (v4.1.1)
[@repeatmasker] were used to create a species-specific repeat database, and mask
those repeats in the genome. All available RNA-seq libraries for *L. rohita*
[@tbl:rnaseq] were downloaded from NCBI and mapped to the masked genome using
hisat2 (v2.1.0) [@doi:10.1038/s41587-019-0201-4]. These alignments were used in
the mikado (v2.0rc2) [@doi:10.1093/gigascience/giy093] and braker2 (v2.1.5)
[@doi:10.1093/nargab/lqaa108] pipelines. Mikado uses the putative transcripts
assembled from the RNA-seq alignments generated via stringtie (v2.1.2)
[@doi:10.1186/s13059-019-1910-1], cufflinks (v2.2.1)
[@doi:10.1038/nprot.2012.016], and trinity (v2.11.0) [@doi:10.1038/nbt.1883];
along with the junction site prediction from portcullis (v1.2.2)
[@doi:10.1093/gigascience/giy131 ], the alignments of the putative transcripts
with UniprotKB Swiss-Prot (v2021.03) [@uniprot], and the ORFs from prodigal
(v2.6.3) [@doi:10.1186/1471-2105-11-119] to select the best representative for
each locus. Braker2 uses the RNA-seq alignments and the gene prediction from
GeneMark-ES (v4.61) [@doi:10.1002/0471250953.bi0406s35] to train a
species-specific Augustus (v3.3.3) [@doi:10.1186/1471-2105-7-62] model. Maker2
(v2.31.10) [@doi:10.1186/1471-2105-12-491] predicts genes based on the new
Augustus, GeneMark, and SNAP models; modifying the predictions based on the
available RNA and protein evidence from the *Cyprinidae* family in the NCBI
RefSeq database. Any predicted genes with an AED above 0.47 were removed from
further analysis. The surviving genes were functionally annotated using
InterProScan (v5.47-82.0) [@doi:10.1093/bioinformatics/btu031] and BLAST+
(v2.9.0) [@doi:10.1186/1471-2105-10-421] alignments against the UniprotKB
Swiss-Prot database. Busco (v5.2.2) [@doi:10.1093/molbev/msab199] was used to
verify the completeness of both the genome and annotations against the
actinopterygii_odb10 database. Lastly, genes spanning large gaps or completely
contained within another gene on the opposite strand were removed using a custom
perl script (https://github.com/IGBB/rohu-genome/).

| BioProject  | BioSample    | Run        | Instrument          | sex            | Tissue        |
|-------------|--------------|------------|---------------------|----------------|---------------|
| PRJNA401304 | SAMN07602342 | SRR6003546 | Illumina HiSeq 2000 | female         | Brain         |
| PRJNA401304 | SAMN07602341 | SRR6003547 | Illumina HiSeq 2000 | female         | Brain         |
| PRJNA401304 | SAMN07602344 | SRR6003548 | Illumina HiSeq 2000 | female         | Pituitary     |
| PRJNA401304 | SAMN07602343 | SRR6003549 | Illumina HiSeq 2000 | female         | Pituitary     |
| PRJNA401304 | SAMN07602346 | SRR6003550 | Illumina HiSeq 2000 | female         | Gonad         |
| PRJNA401304 | SAMN07602345 | SRR6003551 | Illumina HiSeq 2000 | female         | Gonad         |
| PRJNA401304 | SAMN07602348 | SRR6003552 | Illumina HiSeq 2000 | female         | Liver         |
| PRJNA401304 | SAMN07602347 | SRR6003553 | Illumina HiSeq 2000 | female         | Liver         |
| PRJNA449818 | SAMN08918388 | SRR6987066 | NextSeq 500         | female         | Pooled tissue |
| PRJNA449818 | SAMN08918389 | SRR6987067 | NextSeq 500         | male           | Pooled tissue |
| PRJNA449818 | SAMN08918390 | SRR6987068 | NextSeq 500         | female         | whole body    |
| PRJNA528865 | SAMN11246839 | SRR8816555 | Illumina HiSeq 2500 | not applicable | Liver         |
| PRJNA528865 | SAMN11246841 | SRR8816556 | Illumina HiSeq 2500 | not applicable | Liver         |
| PRJNA528865 | SAMN11246840 | SRR8816557 | Illumina HiSeq 2500 | not applicable | Liver         |
| PRJNA450719 | SAMN08944450 | SRR7027730 | NextSeq 500         | female         | Pooled tissue |
| PRJNA450719 | SAMN08944449 | SRR7027731 | NextSeq 500         | male           | Pooled tissue |
| PRJNA450719 | SAMN08944451 | SRR7027732 | NextSeq 500         | male           | Whole body    |

Table: List of SRA accessions used in annotation pipeline. 
A table of all metadata available for these accessions can be found
[here](https://github.com/IGBB/rohu-genome/supp/rna-seq.sra-metadata.csv).
{#tbl:rnaseq}

### Comparitive Genomics

The assenbly statistics, length distributions, BUSCO completeness scorse, and
sequence similarity via dot-plots were compared between the IGBB /Labeo rohita/
genome (reported here) and the two other rohu genomes avaliable on NCBI (CIFA:
GCA\_004120215.1, HRRL: GCA\_017311145.1), as well as all annotated
Cypriniformes genomes from NCBI [@tbl:genomes]. Assembly statistics were
calculated using `abyss-fac` from the ABySS (v2.3.4)
[@doi:10.1101/gr.214346.116]. Lengths distributions were calculated and graphed
using samtools (v1.9) [@doi:10.1093/gigascience/giab008] and the R tidyverse
package [@doi:10.21105/joss.01686]. BUSCO (v5.2.2) [@doi:10.1093/molbev/msab199]
with the actinopterygii\_odb10 database was used to find the BUSCO scores for
each genome. Minimap2 (v2.17-r941) [@doi:10.1093/bioinformatics/bty191] and the
pafCoordsDotPlotly R script (https://github.com/tpoorten/dotPlotly) were used to
create dot-plots. For the Cypriniformes data-set, only chromosome level
assemblies were included in the dot-plots. The /Danio rerio/ and /Triplophysa
tibetana/ genomes were also excluded from the dot-plots since few of the
alignments passed the default quality filter in pafCoordsDotPlotly. The
annotated genes were also compared to all annotated Cypriniformes using
OrthoFinder (v2.5.4) [@doi:10.1186/s13059-019-1832-y].
 
| Organism Scientific Name     | Assembly Name          | Assembly Accession | Level      | Contig N50 | Size       | Submission Date | Gene Count |
|------------------------------|------------------------|--------------------|------------|------------|------------|-----------------|------------|
| Anabarilius grahami          | BGI\_Agra\_1.0         | GCA\_003731715.1   | Scaffold   | 36058      | 991887266  | 2018-11-15      | 23906      |
| Carassius auratus            | ASM336829v1            | GCF\_003368295.1   | Chromosome | 821153     | 1820618472 | 2018-08-09      | 83650      |
| Cyprinus carpio              | ASM1834038v1           | GCF\_018340385.1   | Chromosome | 1558716    | 1680118328 | 2021-05-12      | 59559      |
| Danionella translucida       | ASM722483v1            | GCA\_007224835.1   | Scaffold   | 133131     | 735303417  | 2019-07-22      | 35803      |
| Danio rerio                  | GRCz11                 | GCF\_000002035.6   | Chromosome | 1422317    | 1373454788 | 2017-05-09      | 40031      |
| Onychostoma macrolepis       | ASM1243209v1           | GCA\_012432095.1   | Chromosome | 10809623   | 886569841  | 2020-04-17      | 24754      |
| Pimephales promelas          | EPA\_FHM\_2.0          | GCA\_016745375.1   | Scaffold   | 295770     | 1066412313 | 2021-01-24      | 26150      |
| Puntigrus tetrazona          | ASM1883169v1           | GCF\_018831695.1   | Chromosome | 1423374    | 730801986  | 2021-06-10      | 40303      |
| Sinocyclocheilus anshuiensis | SAMN03320099.WGS\_v1.1 | GCF\_001515605.1   | Scaffold   | 17271      | 1632701648 | 2015-12-14      | 52005      |
| Sinocyclocheilus grahami     | SAMN03320097.WGS\_v1.1 | GCF\_001515645.1   | Scaffold   | 29354      | 1750271176 | 2015-12-16      | 55200      |
| Sinocyclocheilus rhinocerous | SAMN03320098\_v1.1     | GCF\_001515625.1   | Scaffold   | 18758      | 1655769822 | 2015-12-14      | 53875      |
| Triplophysa tibetana         | ASM836982v1            | GCA\_008369825.1   | Chromosome | 2833080    | 652930317  | 2019-09-12      | 24398      |

Table: List of Cypriniformes genomes used in comparitive analyses
{#tbl:genomes}

### ddRAD-Seq & SNP Discovery 

#### WORLDFISH â€“ Fin clipping collection 

#### ZENAIDA â€“ ddRAD-Seq method 

#### CORRINNE â€“ SNP Discovery, population analyses, interested in using the gender, which contigs associated with sex 

### Data Availability 

The data used for the *L. rohita* genome and annotation are available at NCBI
under the BioProject PRJNA650519. The assembled genome sequence and annotations
are available at GenBank under accessions JACTAM000000000. The raw data is
available at the SRA (Sequence Read Archive) under accessions SRR12580210 â€“
SRR12580221.


## Results & Discussion 

<!-- 
Results and Discussion: Should include a description of the genomes, coverage 
amount of genome sequenced, details of sequence variants ascertained, 
identifications of deletions, comparison of strains/genomes (if reporting WGS 
of multiple strains/organisms), and relevance of findings to overall 
organism/strain catalog.
-->

### Sequencing & Assembly 

The C-value of *L. rohita* was previously reported as 1.99 pg (~1.95Gb) using Feulgen densitometry [@genome-size] or 1.5Gb using k-mer estimation [@pmcid:PMC7186481]. 
However, the flow cytometry results (Table [@tbl:flowcyto]) show a C-value of 0.99 pg (~0.97Gb) with a standard deviation of 0.02 across all measurements. 
The smaller C-value is also closer to the genome estimate produced by GenomeScope (0.97Gb) and the final genome assembly size of 0.95 Gb.


| Specimen Name     | Number of Sample nuclei | Average sample fluorescence | Number of standard nuclei | Average standard fluorescence | Estimated Genome size ^2^ | HAPLOID       |
|-------------------|-------------------------|-----------------------------|---------------------------|-------------------------------|---------------------------|---------------|
| Fish 1 Sample 1   | 16020                   | 27350                       | 2065                      | 69247                         | 2.049857756               | 1.024928878   |
| Fish 2 Sample 1    | 13082                   | 25929                       | 6570                      | 66671                         | 2.018441451               | 1.0092207255  |
| Fish 2 Sample 2   | 15402                   | 25665                       | 4354                      | 67489                         | 1.973674969               | 0.9868374845  |
| Fish 3 Sample 1   | 15124                   | 25798                       | 4442                      | 68195                         | 1.963364176               | 0.981682088   |
| Fish 3 Sample 2   | 14923                   | 25763                       | 4823                      | 68837                         | 1.942414254               | 0.971207127   |
| Fish 4 Sample 1   | 13320                   | 26346                       | 5913                      | 69665                         | 1.962760927               | 0.9813804635  |
| Fish 4 Sample 2   | 5624                    | 26612                       | 4097                      | 68876                         | 2.005288925               | 1.0026444625  |
| Fish 5 Sample 1   | 6771                    | 25761                       | 3080                      | 68825                         | 1.942602107               | 0.9713010535  |
| Fish 5 Sample 2   | 15926                   | 26369                       | 3352                      | 68832                         | 1.988248344               | 0.994124172   |
| Standard only ^1^ | 3                       | 25258                       | 3311                      | 64331                         | NA                        |               |
|                   |                         |                             |                           | Average                       | 1.982961434               | 0.9914807172  |
|                   |                         |                             |                           | Standard Deviation            | 0.03607582                | 0.01803790999 |

Table: Flow cytometry results for 5 *L. rohita* blood samples, measured twice. 
1) Trout erythrocyte nuclei: Genome size = 5.19pg.
2) Genome estimate calculated as (average sample fluorescence/ average standard fluorescence * standard genome size) in picograms DIPLOID (i.e., 2C).
{#tbl:flowcyto}

A total of 130.5 Gb of Nanopore long reads from 44.7 million read, and 261 Gb of Illumina short reads from 870 million pairs were produced, along with 382 million pairs (114 Gb) for the Hi-C library.
The initial *de novo* assembly consisted of 4999 contigs with an N50 of 1.28 Mb. 
After the Bionano and HiC data was incorporated, the total number of sequences dropped to 2899 and the N50 increased to 29.9 Mb.
The final assembly consisted of 25 chromosome length scaffolds and 2844 unplaced scaffolds, ranging in size from 1,479bp to 7.18 Mb.
The chromosome scaffold consist of one to eight sequences with all but three made from three or fewer sequences.
Table [@tbl:assembly] contains a common assembly statistics for each step.
The final genome size is 97.9% of the estimated genome size. 
The annotation pipeline produced 51,079 primary transcripts, 31,274 surviving the AED, gap, and overlapping filters. 
BUSCO analysis show the genome completely contains 98.1% of the 3640 orthologs in the actinopterygii_odb10 database with 37 (1%) duplicated;
while, the filtered transcriptome contains 84.5% of the total orthologs complete with 74 duplicated.
A complete comparison of the BUSCO analyses can be found in Table [@tbl:busco].

| n    | n:500 | L50 | min  | N75      | N50      | N25      | E-size   | max      | sum      | name    |
|------|-------|-----|------|----------|----------|----------|----------|----------|----------|---------|
| 4999 | 4999  | 202 | 1348 | 514919   | 1281850  | 2395030  | 1727184  | 7832582  | 9.43E+08 | wtdbg2  |
| 3709 | 3706  | 15  | 1479 | 1.13E+07 | 2.65E+07 | 3.08E+07 | 2.20E+07 | 3.79E+07 | 9.46E+08 | bionano |
| 2899 | 2896  | 14  | 1479 | 2.64E+07 | 2.99E+07 | 3.43E+07 | 2.69E+07 | 4.45E+07 | 9.46E+08 | hic     |
| 2872 | 2869  | 13  | 1479 | 2.88E+07 | 3.25E+07 | 3.61E+07 | 3.00E+07 | 4.53E+07 | 9.46E+08 | ragtag  |

Table: Assembly statistics for each stage of the assembly
{#tbl:assembly}

| Type                                | Genome | Unfiltered Transcriptome | Filtered Transcriptome |
|-------------------------------------|--------|--------------------------|------------------------|
| Complete BUSCOs (C)                 | 3571   | 3139                     | 3078                   |
| Complete and single-copy BUSCOs (S) | 3534   | 3064                     | 3001                   |
| Complete and duplicated BUSCOs (D)  | 37     | 75                       | 74                     |
| Fragmented BUSCOs (F)               | 23     | 192                      | 170                    |
| Missing BUSCOs (M)                  | 46     | 309                      | 392                    |
| Total BUSCO groups searched         | 3640   | 3640                     | 3640                   |

Table: BUSCO analysis for the genome and transcriptome, before and after AED filtering.
{#tbl:busco}

```
================================================== 
file name: Rohu.genome.fa            
sequences:          2872 
total length: 1128029156 bp (945637473 bp excl N/X-runs) 
GC level:         36.05 % 
bases masked:  465289941 bp (41.25 %) 
================================================== 
               number of      length   percentage 
               elements*    occupied  of sequence 
-------------------------------------------------- 
Retroelements       123482     53424977 bp    4.74 % 
   SINEs:                0            0 bp    0.00 % 
   Penelope              0            0 bp    0.00 % 
   LINEs:            29838     13842585 bp    1.23 % 
    CRE/SLACS            0            0 bp    0.00 % 
     L2/CR1/Rex      15537      8265099 bp    0.73 % 
     R1/LOA/Jockey     981       643697 bp    0.06 % 
     R2/R4/NeSL        936       732043 bp    0.06 % 
     RTE/Bov-B         271       181676 bp    0.02 % 
     L1/CIN4          7978      1801119 bp    0.16 % 
   LTR elements:     93644     39582392 bp    3.51 % 
     BEL/Pao          1287      1138420 bp    0.10 % 
     Ty1/Copia         221       110739 bp    0.01 % 
     Gypsy/DIRS1     32714     19073624 bp    1.69 % 
       Retroviral     2443      1815904 bp    0.16 % 
  
DNA transposons      79815     22485920 bp    1.99 % 
   hobo-Activator     2242       639067 bp    0.06 % 
   Tc1-IS630-Pogo    49209     16661924 bp    1.48 % 
   En-Spm                0            0 bp    0.00 % 
   MuDR-IS905            0            0 bp    0.00 % 
   PiggyBac              0            0 bp    0.00 % 
   Tourist/Harbinger  1794       259934 bp    0.02 % 
   Other (Mirage,      900       209002 bp    0.02 % 
    P-element, Transib) 
  
Rolling-circles        196        71055 bp    0.01 % 
  
Unclassified:       1774260    361869982 bp   32.08 % 
  
Total interspersed repeats:   437780879 bp   38.81 % 
  
  
Small RNA:               0            0 bp    0.00 % 
  
Satellites:              1          267 bp    0.00 % 
Simple repeats:     504151     23839806 bp    2.11 % 
Low complexity:      58354      3597934 bp    0.32 % 
================================================== 
  
* most repeats fragmented by insertions or deletions 
  have been counted as one element 
                                                       
  
RepeatMasker version 4.1.1 , default mode 
                                         
run with rmblastn version 2.10.0+ 
The query was compared to classified sequences in "Rohu-families.fa" 
 
```

### Comparative genomics

#### Labeo Rohita


| Name | n     | n:500 | L50 | min  | N75     | N50     | N25     | E-size  | max     | sum     |
|------|-------|-------|-----|------|---------|---------|---------|---------|---------|---------|
| CIFA | 13623 | 13617 | 182 | 500  | 774698  | 2006958 | 4279019 | 2905260 | 15.15e6 | 1.427e9 |
| HRRL | 13660 | 13650 | 182 | 500  | 774698  | 2006958 | 4279019 | 2905216 | 15.15e6 | 1.427e9 |
| IGBB | 2869  | 2869  | 13  | 1479 | 28.79e6 | 32.5e6  | 36.06e6 | 30.03e6 | 45.29e6 | 945.5e6 |

Table: Assembly stats for rohu genomes
{#tbl:assembly-stats}

![
Length distribution of scaffolds for each rohu assembly.
Each bar represents the number of bases contained in scaffolds of a particular length.
](../comp-gen/1-stats/length.rohu.hires.png "Length distro")[#fig:len-distro]

![
BUSCO score for each rohu genome
](../comp-gen/1-busco/rohu/busco.summary.png "Rohu BUSCOs")[#fig:busco-rohu]

![
Dot-plots comparing sequence similarity between rohu genomes
](../comp-gen/2-dotplot/rohu.png)[#fig:dotplot-rohu]

#### Cypriniformes

![
BUSCO score for each rohu genome
](../comp-gen/1-busco/cypriniformes/busco.summary.png "Rohu BUSCOs")[#fig:busco-carps]

![
Dot-plots comparing sequence similarity between rohu genomes
](../comp-gen/2-dotplot/cypriniformes.rm-uninformative.png)[#fig:dotplot-carps]


|                              | Number of Genes | Genes in Orthogroups | Unassigned Genes | Orthogroups Containing Species | Species-specific Orthogroups | Genes in Species-specific Orthogroups |
|------------------------------|-----------------|----------------------|------------------|--------------------------------|------------------------------|---------------------------------------|
| Anabarilius grahami          | 23906           | 21942 (91.8%)        | 1964 (8.2%)      | 15217 (51.5%)                  | 180                          | 888 (3.7%)                            |
| Carassius auratus            | 96703           | 93988 (97.2%)        | 2715 (2.8%)      | 21896 (74.2%)                  | 472                          | 1775 (1.8%)                           |
| Cyprinus carpio              | 80686           | 78684 (97.5%)        | 2002 (2.5%)      | 21584 (73.1%)                  | 287                          | 964 (1.2%)                            |
| Danio rerio                  | 52829           | 51951 (98.3%)        | 878  (1.7%)      | 20671 (70.0%)                  | 392                          | 2184 (4.1%)                           |
| Danionella translucida       | 35381           | 32943 (93.1%)        | 2438 (6.9%)      | 19073 (64.6%)                  | 540                          | 1941 (5.5%)                           |
| Labeo rohita                 | 31274           | 29904 (95.6%)        | 1370 (4.4%)      | 18740 (63.5%)                  | 161                          | 1581 (5.1%)                           |
| Onychostoma macrolepis       | 24754           | 24483 (98.9%)        | 271  (1.1%)      | 19276 (65.3%)                  | 137                          | 603 (2.4%)                            |
| Pimephales promelas          | 47578           | 45412 (95.4%)        | 2166 (4.6%)      | 19884 (67.4%)                  | 506                          | 1826 (3.8%)                           |
| Puntigrus tetrazona          | 48681           | 48094 (98.8%)        | 587  (1.2%)      | 20582 (69.7%)                  | 129                          | 517 (1.1%)                            |
| Sinocyclocheilus anshuiensis | 68474           | 66456 (97.1%)        | 2018 (2.9%)      | 21485 (72.8%)                  | 114                          | 344 (0.5%)                            |
| Sinocyclocheilus grahami     | 67410           | 63316 (93.9%)        | 4094 (6.1%)      | 22326 (75.6%)                  | 338                          | 793 (1.2%)                            |
| Sinocyclocheilus rhinocerous | 68562           | 65831 (96.0%)        | 2731 (4.0%)      | 21884 (74.1%)                  | 172                          | 414 (0.6%)                            |
| Triplophysa tibetana         | 24310           | 23279 (95.8%)        | 1031 (4.2%)      | 18734 (63.5%)                  | 125                          | 480 (2.0%)                            |

Table: Summary statistics per species from OrthoFinder.
{#tbl:orthofinder}

_SNP discovery and population similarities among rohu fisheries_

_Labeo rohita_ is an important aquaculture fish in many areas of south Asia [I
donâ€™t know how to put in this reference FAO (2019) Food and Agricultural
Organization. Yearbook of Fishery and Aquaculture Statistics 2017].

The high fecundity and relative ease of cultivation has made rohu successful in
breeding programs, leading to the generation of a fast-growing strain called
Jayanti Historically, rohu breeding and research has relied on low-throughput
information, such as cloning individual genes of interest (reviewed in
[@doi:10.1111/raq.12444]) or microsatellite-based selective breeding
[@doi:10.1007/s10499-016-0104-z]; however, recent years have seen increased
interest in applying high-throughput methods to rohu breeding, such as SNP
panels [@doi:10.1186/1471-2164-15-541] [@doi:10.1016/J.GENREP.2016.12.004],
genotyping-by-sequencing [@doi:10.3389/fgene.2019.00597], transcriptomics
[@doi:10.1186/1471-2164-15-541] [@doi:10.1080/10495398.2019.1707683], and
degradomics [@doi:10.1016/j.ygeno.2019.07.010]. Recently, a draft genome was
published for L. rohita along with pooled resequencing from three individuals in
10 populations [@doi:10.3389/fgene.2020.00386]; however, genetic diversity and
divergence among the populations was not reported.















## Acknowledgements

ISU Office of Biotechnology Flow Cytometry Facility 

ResearchIT Iowa State University 


## References {.page_break_before}

<!-- Explicitly insert bibliography here -->
<div id="refs"></div>

[@genome-size]: http://zotero.org/users/local/gVY292am/items/GYK6YE2I
