<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Mark A. Arick II" />
  <meta name="author" content="Chuan-Yu Hsu" />
  <meta name="author" content="Zenaida Magbanua" />
  <meta name="author" content="Corrinne E. Grover" />
  <meta name="author" content="Emma R. Miller" />
  <meta name="author" content="Olga Pechanova" />
  <meta name="author" content="Adam Thrash" />
  <meta name="author" content="Ramey C. Youngblood" />
  <meta name="author" content="Lauren Ezzell" />
  <meta name="author" content="Samsul Alam" />
  <meta name="author" content="John Benzie" />
  <meta name="author" content="Matthew Hamilton" />
  <meta name="author" content="Attila Karsi" />
  <meta name="author" content="Mark L. Lawrence" />
  <meta name="author" content="Daniel G. Peterson" />
  <meta name="dcterms.date" content="2022-02-15" />
  <meta name="keywords" content="genome sequencing, ddrad sequencing, nanopore, Labeo rohita" />
  <title>A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta name="citation_title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta property="og:title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta property="twitter:title" content="A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers" />
  <meta name="dc.date" content="2022-02-15" />
  <meta name="citation_publication_date" content="2022-02-15" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Mark A. Arick II" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="0000-0002-7207-3052" />
  <meta name="citation_author" content="Chuan-Yu Hsu" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Zenaida Magbanua" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Corrinne E. Grover" />
  <meta name="citation_author_institution" content="Ecology, Evolution, and Organismal Biology Dept., Iowa State University, Ames, IA, 50010 USA" />
  <meta name="citation_author_orcid" content="0000-0003-3878-5459" />
  <meta name="citation_author" content="Emma R. Miller" />
  <meta name="citation_author_institution" content="Ecology, Evolution, and Organismal Biology Dept., Iowa State University, Ames, IA, 50010 USA" />
  <meta name="citation_author_orcid" content="0000-0001-9009-5303" />
  <meta name="citation_author" content="Olga Pechanova" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Adam Thrash" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="0000-0002-4097-7663" />
  <meta name="citation_author" content="Ramey C. Youngblood" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Lauren Ezzell" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Samsul Alam" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="John Benzie" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Matthew Hamilton" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Attila Karsi" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Mark L. Lawrence" />
  <meta name="citation_author_institution" content="placeholder" />
  <meta name="citation_author_orcid" content="XXXX-XXXX-XXXX-XXXX" />
  <meta name="citation_author" content="Daniel G. Peterson" />
  <meta name="citation_author_institution" content="Institute for Genomics, Biocomputing &amp; Biotechnology, Mississippi State University, MS, 39762 USA" />
  <meta name="citation_author_orcid" content="0000-0002-0274-5968" />
  <link rel="canonical" href="https://IGBB.github.io/rohu-genome/" />
  <meta property="og:url" content="https://IGBB.github.io/rohu-genome/" />
  <meta property="twitter:url" content="https://IGBB.github.io/rohu-genome/" />
  <meta name="citation_fulltext_html_url" content="https://IGBB.github.io/rohu-genome/" />
  <meta name="citation_pdf_url" content="https://IGBB.github.io/rohu-genome/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://IGBB.github.io/rohu-genome/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/" />
  <meta name="manubot_html_url_versioned" content="https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/" />
  <meta name="manubot_pdf_url_versioned" content="https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">A high-quality chromosome-level genome assembly of rohu carp, Labeo rohita, and discovery of SNP markers</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://IGBB.github.io/rohu-genome/v/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72/">permalink</a>)
was automatically generated
from <a href="https://github.com/IGBB/rohu-genome/tree/5350e6f1612cdf80dc3f8355a6eb4f5b75ca2a72">IGBB/rohu-genome@5350e6f</a>
on February 15, 2022.
</em></small></p>
<h2 id="authors">Authors</h2>
<p>Mark A. Arick II<sup>1
<a href="https://orcid.org/0000-0002-7207-3052"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Chuan-Yu Hsu<sup>1
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Zenaida Magbanua<sup>1
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Corrinne E. Grover<sup>2
<a href="https://orcid.org/0000-0003-3878-5459"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Emma R. Miller<sup>2
<a href="https://orcid.org/0000-0001-9009-5303"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Olga Pechanova<sup>1
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Adam Thrash<sup>1
<a href="https://orcid.org/0000-0002-4097-7663"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Ramey C. Youngblood<sup>1
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Lauren Ezzell<sup>1
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Samsul Alam<sup>3
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
John Benzie<sup>3
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Matthew Hamilton<sup>3
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Attila Karsi<sup>3
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Mark L. Lawrence<sup>3
<a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup>,
Daniel G. Peterson<sup>1
<a href="https://orcid.org/0000-0002-0274-5968"><img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" /></a>
</sup></p>
<p><small></p>
<p></small></p>
<h2 id="abstract">Abstract</h2>
<!--  Abstract: ~250 words. Concise summary of the results. -->
<h2 id="introduction">Introduction</h2>
<!--  Introduction: no more than 500 words. The introduction should discuss the rationale for the WGS of the particular organism/strain. -->
<h2 id="methods-materials">Methods &amp; Materials</h2>
<!--  
Methods & Materials: should include the following (note: author may write 
additional subheadings as appropriate – the below points are not designed 
to be a list of complete subheadings, but rather, topics to be covered).
 -  Organism/Strain origin and derivation 
 -  Sequencing methods and preparation details
 -  Data processing methods 
 -  Data availability listing where to access all raw reads, processed data, 
    BWTs, and other relevant data. The assembly should be uploaded to NCBI or 
    ENA.
 -->
<h3 id="genome-sequencing-flowcytometry">Genome Sequencing &amp; Flowcytometry</h3>
<p>ALAM – Rohu blood collection</p>
<h4 id="flow-cytometry">Flow cytometry</h4>
<p>The genome size of <em>Labeo rohita</em> was estimated for five samples via flow
cytometry using trout erythrocyte nuclei (TENs;
https://www.biosure.com/tens.html) as a standard (genome size=6.5pg). For each
sample, nuclei were stabilized in 200 ul of LB01-propidium iodide (PI) buffer as
per <span class="citation" data-cites="11lZmTCLG">[<a href="#ref-11lZmTCLG" role="doc-biblioref">1</a>]</span>, and two drops of TENs standard were used per 50ul of fish
blood. Each sample was measured twice, totaling 10 runs overall. Only
measurements with greater than 5,000 nuclei and a coefficient of variation (CV)
of less than 3% were retained <span class="citation" data-cites="11lZmTCLG">[<a href="#ref-11lZmTCLG" role="doc-biblioref">1</a>]</span>.</p>
<h4 id="illumina-short-read-sequencing">Illumina short-read sequencing</h4>
<p>A total of 2 µg of extracted genomic DNA was used for DNA-Seq library
preparation using Illumina TruSeq DNA PCR-free Library Prep Kit (Illumina, San
Diego, CA, USA). The final DNA-Seq library with the insert size range of 350 bp
to 450 bp was submitted to Novogene company (www.en.novogene.com) for total of 2
lanes of PE150 Illumina HiSeq X-Ten (Illumina, San Diego, CA, USA) sequencing.</p>
<h4 id="oxford-nanopore-sequencing">Oxford Nanopore sequencing</h4>
<p>For each Nanopore sequencing run, 2 to 2.5 µg of genomic DNA was used in the
library preparation with Nanopore Genomic DNA Ligation Sequencing Kit SQK-LSK
109 (Oxford Nanopore Technologies, Oxford, UK). The final library (about 700 to
750 ng) was loaded on a Nanopore Flow Cell R9.4.1 (Oxford Nanopore Technologies,
Oxford, UK) and sequenced on GridION sequencer (Oxford Nanopore Technologies,
Oxford, UK) for 48 hours. A total of 10 flow cell runs were conducted for the
genome assembly.</p>
<h4 id="hi-c-sequencing">Hi-C sequencing</h4>
<p>One hundred µl of fish blood was subjected to the Hi-C library preparation using
the Proximo Hi-C Animal Kit (Phase Genomics, Seattle, WA, USA). The final
DNA-Seq library was submitted to Novogene (www.en.novogene.com) for 1 lane of
PE150 Illumina HiSeq X-Ten (Illumina, San Diego, CA, USA) sequencing.</p>
<h4 id="bionano">Bionano</h4>
<h3 id="assembly-annotation">Assembly &amp; Annotation</h3>
<p>Jellyfish (v2.2.10) <span class="citation" data-cites="14dfsRv4c">[<a href="#ref-14dfsRv4c" role="doc-biblioref">2</a>]</span> and GenomeScope (v1.0)
<span class="citation" data-cites="1H1zfojOz">[<a href="#ref-1H1zfojOz" role="doc-biblioref">3</a>]</span> estimated the genome size using the Illumina paired reads
digested into 50-mers.</p>
<p>Nanopore data was filtered to remove the control lambda-phage and sequences
shorter than 1000 bases using the nanopack tool suite (v1.0.1)
<span class="citation" data-cites="88NBy0">[<a href="#ref-88NBy0" role="doc-biblioref">4</a>]</span>. Trimmomatic (v0.32)
<span class="citation" data-cites="l4tjRGIB">[<a href="#ref-l4tjRGIB" role="doc-biblioref">5</a>]</span> removed adapters, trimmed low quality
bases, and filtered reads shorter than 85bp. The filtered nanopore data was
assembled into contigs using wtdbg2 (v2.4) <span class="citation" data-cites="9l1p27cN">[<a href="#ref-9l1p27cN" role="doc-biblioref">6</a>]</span>. The
contigs were polished using the two iterations of racon (v1.4.0)
<span class="citation" data-cites="zY4KP2bZ">[<a href="#ref-zY4KP2bZ" role="doc-biblioref">7</a>]</span> with minimap2 (v2.17)
(<span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q" role="doc-biblioref">8</a>]</span>) mapping the nanopore reads. The contigs
were further polished using pilon (v1.23) <span class="citation" data-cites="LDP9ZoBM">[<a href="#ref-LDP9ZoBM" role="doc-biblioref">9</a>]</span>
with bwa (v0.7.10) <span class="citation" data-cites="roi5bwXL">[<a href="#ref-roi5bwXL" role="doc-biblioref">10</a>]</span> mapping the Illumina
paired reads. The contigs were scaffolded using Bionano Solve (
Solve3.4.1_09262019)
[<span class="citation" data-cites="wgWE6D4o">[<a href="#ref-wgWE6D4o" role="doc-biblioref">11</a>]</span>/) and
SALSA (v2.3) <span class="citation" data-cites="KEqlQin">[<a href="#ref-KEqlQin" role="doc-biblioref">12</a>]</span>. That scaffolds larger than
10Mb were linked and oriented based on the Onychostoma macrolepis genome
<span class="citation" data-cites="GNj8xY3t">[<a href="#ref-GNj8xY3t" role="doc-biblioref">13</a>]</span>, the most similar chromosome assembly available
on NCBI, using ragtag (v1.1.1) <span class="citation" data-cites="zePiKkRw">[<a href="#ref-zePiKkRw" role="doc-biblioref">14</a>]</span>.</p>
<p>RepeatModeler (v2.0.1) <span class="citation" data-cites="DUW8OYnu">[<a href="#ref-DUW8OYnu" role="doc-biblioref">15</a>]</span> and RepeatMasker (v4.1.1)
<span class="citation" data-cites="repeatmasker">[<a href="#ref-repeatmasker" role="doc-biblioref"><strong>repeatmasker?</strong></a>]</span> were used to create a species-specific repeat database, and mask
those repeats in the genome. All available RNA-seq libraries for <em>L. rohita</em>
[<a href="#tbl:rnaseq">1</a>] were downloaded from NCBI and mapped to the masked genome using
hisat2 (v2.1.0) <span class="citation" data-cites="1GLBRcU8y">[<a href="#ref-1GLBRcU8y" role="doc-biblioref">16</a>]</span>. These alignments were used in
the mikado (v2.0rc2) <span class="citation" data-cites="Nc6oNbiA">[<a href="#ref-Nc6oNbiA" role="doc-biblioref">17</a>]</span> and braker2 (v2.1.5)
<span class="citation" data-cites="Sq0xBB4s">[<a href="#ref-Sq0xBB4s" role="doc-biblioref">18</a>]</span> pipelines. Mikado uses the putative transcripts
assembled from the RNA-seq alignments generated via stringtie (v2.1.2)
<span class="citation" data-cites="1F8s8oDfw">[<a href="#ref-1F8s8oDfw" role="doc-biblioref">19</a>]</span>, cufflinks (v2.2.1)
<span class="citation" data-cites="14kgcZI3s">[<a href="#ref-14kgcZI3s" role="doc-biblioref">20</a>]</span>, and trinity (v2.11.0) <span class="citation" data-cites="oU9Lbgmd">[<a href="#ref-oU9Lbgmd" role="doc-biblioref">21</a>]</span>;
along with the junction site prediction from portcullis (v1.2.2)
<span class="citation" data-cites="B9kt43k8">[<a href="#ref-B9kt43k8" role="doc-biblioref">22</a> ]</span>, the alignments of the putative transcripts
with UniprotKB Swiss-Prot (v2021.03) <span class="citation" data-cites="uniprot">[<a href="#ref-uniprot" role="doc-biblioref"><strong>uniprot?</strong></a>]</span>, and the ORFs from prodigal
(v2.6.3) <span class="citation" data-cites="lX665mdh">[<a href="#ref-lX665mdh" role="doc-biblioref">23</a>]</span> to select the best representative for
each locus. Braker2 uses the RNA-seq alignments and the gene prediction from
GeneMark-ES (v4.61) <span class="citation" data-cites="7OpVD07l">[<a href="#ref-7OpVD07l" role="doc-biblioref">24</a>]</span> to train a
species-specific Augustus (v3.3.3) <span class="citation" data-cites="5gYwalTK">[<a href="#ref-5gYwalTK" role="doc-biblioref">25</a>]</span> model. Maker2
(v2.31.10) <span class="citation" data-cites="UdSgHXyu">[<a href="#ref-UdSgHXyu" role="doc-biblioref">26</a>]</span> predicts genes based on the new
Augustus, GeneMark, and SNAP models; modifying the predictions based on the
available RNA and protein evidence from the <em>Cyprinidae</em> family in the NCBI
RefSeq database. Any predicted genes with an AED above 0.47 were removed from
further analysis. The surviving genes were functionally annotated using
InterProScan (v5.47-82.0) <span class="citation" data-cites="N1v57w5y">[<a href="#ref-N1v57w5y" role="doc-biblioref">27</a>]</span> and BLAST+
(v2.9.0) <span class="citation" data-cites="nEsJGUWa">[<a href="#ref-nEsJGUWa" role="doc-biblioref">28</a>]</span> alignments against the UniprotKB
Swiss-Prot database. Busco (v5.2.2) <span class="citation" data-cites="Vq0QVho4">[<a href="#ref-Vq0QVho4" role="doc-biblioref">29</a>]</span> was used to
verify the completeness of both the genome and annotations against the
actinopterygii_odb10 database. Lastly, genes spanning large gaps or completely
contained within another gene on the opposite strand were removed using a custom
perl script (https://github.com/IGBB/rohu-genome/).</p>
<div id="tbl:rnaseq" class="tablenos">
<table id="tbl:rnaseq">
<caption><span>Table 1:</span> List of SRA accessions used in annotation pipeline.
A table of all metadata available for these accessions can be found
<a href="https://github.com/IGBB/rohu-genome/supp/rna-seq.sra-metadata.csv">here</a>.
</caption>
<thead>
<tr class="header">
<th>BioProject</th>
<th>BioSample</th>
<th>Run</th>
<th>Instrument</th>
<th>sex</th>
<th>Tissue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PRJNA401304</td>
<td>SAMN07602342</td>
<td>SRR6003546</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Brain</td>
</tr>
<tr class="even">
<td>PRJNA401304</td>
<td>SAMN07602341</td>
<td>SRR6003547</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Brain</td>
</tr>
<tr class="odd">
<td>PRJNA401304</td>
<td>SAMN07602344</td>
<td>SRR6003548</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Pituitary</td>
</tr>
<tr class="even">
<td>PRJNA401304</td>
<td>SAMN07602343</td>
<td>SRR6003549</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Pituitary</td>
</tr>
<tr class="odd">
<td>PRJNA401304</td>
<td>SAMN07602346</td>
<td>SRR6003550</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Gonad</td>
</tr>
<tr class="even">
<td>PRJNA401304</td>
<td>SAMN07602345</td>
<td>SRR6003551</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Gonad</td>
</tr>
<tr class="odd">
<td>PRJNA401304</td>
<td>SAMN07602348</td>
<td>SRR6003552</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Liver</td>
</tr>
<tr class="even">
<td>PRJNA401304</td>
<td>SAMN07602347</td>
<td>SRR6003553</td>
<td>Illumina HiSeq 2000</td>
<td>female</td>
<td>Liver</td>
</tr>
<tr class="odd">
<td>PRJNA449818</td>
<td>SAMN08918388</td>
<td>SRR6987066</td>
<td>NextSeq 500</td>
<td>female</td>
<td>Pooled tissue</td>
</tr>
<tr class="even">
<td>PRJNA449818</td>
<td>SAMN08918389</td>
<td>SRR6987067</td>
<td>NextSeq 500</td>
<td>male</td>
<td>Pooled tissue</td>
</tr>
<tr class="odd">
<td>PRJNA449818</td>
<td>SAMN08918390</td>
<td>SRR6987068</td>
<td>NextSeq 500</td>
<td>female</td>
<td>whole body</td>
</tr>
<tr class="even">
<td>PRJNA528865</td>
<td>SAMN11246839</td>
<td>SRR8816555</td>
<td>Illumina HiSeq 2500</td>
<td>not applicable</td>
<td>Liver</td>
</tr>
<tr class="odd">
<td>PRJNA528865</td>
<td>SAMN11246841</td>
<td>SRR8816556</td>
<td>Illumina HiSeq 2500</td>
<td>not applicable</td>
<td>Liver</td>
</tr>
<tr class="even">
<td>PRJNA528865</td>
<td>SAMN11246840</td>
<td>SRR8816557</td>
<td>Illumina HiSeq 2500</td>
<td>not applicable</td>
<td>Liver</td>
</tr>
<tr class="odd">
<td>PRJNA450719</td>
<td>SAMN08944450</td>
<td>SRR7027730</td>
<td>NextSeq 500</td>
<td>female</td>
<td>Pooled tissue</td>
</tr>
<tr class="even">
<td>PRJNA450719</td>
<td>SAMN08944449</td>
<td>SRR7027731</td>
<td>NextSeq 500</td>
<td>male</td>
<td>Pooled tissue</td>
</tr>
<tr class="odd">
<td>PRJNA450719</td>
<td>SAMN08944451</td>
<td>SRR7027732</td>
<td>NextSeq 500</td>
<td>male</td>
<td>Whole body</td>
</tr>
</tbody>
</table>
</div>
<h3 id="comparitive-genomics">Comparitive Genomics</h3>
<p>The assenbly statistics, length distributions, BUSCO completeness scorse, and
sequence similarity via dot-plots were compared between the IGBB /Labeo rohita/
genome (reported here) and the two other rohu genomes avaliable on NCBI (CIFA:
GCA_004120215.1, HRRL: GCA_017311145.1), as well as all annotated
Cypriniformes genomes from NCBI [<a href="#tbl:genomes">2</a>]. Assembly statistics were
calculated using <code>abyss-fac</code> from the ABySS (v2.3.4)
<span class="citation" data-cites="2W9THYfa">[<a href="#ref-2W9THYfa" role="doc-biblioref">30</a>]</span>. Lengths distributions were calculated and graphed
using samtools (v1.9) <span class="citation" data-cites="eyOvxHVn">[<a href="#ref-eyOvxHVn" role="doc-biblioref">31</a>]</span> and the R tidyverse
package <span class="citation" data-cites="EWll2pyF">[<a href="#ref-EWll2pyF" role="doc-biblioref">32</a>]</span>. BUSCO (v5.2.2) <span class="citation" data-cites="Vq0QVho4">[<a href="#ref-Vq0QVho4" role="doc-biblioref">29</a>]</span>
with the actinopterygii_odb10 database was used to find the BUSCO scores for
each genome. Minimap2 (v2.17-r941) <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q" role="doc-biblioref">8</a>]</span> and the
pafCoordsDotPlotly R script (https://github.com/tpoorten/dotPlotly) were used to
create dot-plots. For the Cypriniformes data-set, only chromosome level
assemblies were included in the dot-plots. The /Danio rerio/ and /Triplophysa
tibetana/ genomes were also excluded from the dot-plots since few of the
alignments passed the default quality filter in pafCoordsDotPlotly. The
annotated genes were also compared to all annotated Cypriniformes using
OrthoFinder (v2.5.4) <span class="citation" data-cites="1E6awuyBZ">[<a href="#ref-1E6awuyBZ" role="doc-biblioref">33</a>]</span>.</p>
<div id="tbl:genomes" class="tablenos">
<table id="tbl:genomes">
<caption><span>Table 2:</span> List of Cypriniformes genomes used in comparitive analyses
</caption>
<colgroup>
<col style="width: 21%" />
<col style="width: 17%" />
<col style="width: 14%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 12%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="header">
<th>Organism Scientific Name</th>
<th>Assembly Name</th>
<th>Assembly Accession</th>
<th>Level</th>
<th>Contig N50</th>
<th>Size</th>
<th>Submission Date</th>
<th>Gene Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Anabarilius grahami</td>
<td>BGI_Agra_1.0</td>
<td>GCA_003731715.1</td>
<td>Scaffold</td>
<td>36058</td>
<td>991887266</td>
<td>2018-11-15</td>
<td>23906</td>
</tr>
<tr class="even">
<td>Carassius auratus</td>
<td>ASM336829v1</td>
<td>GCF_003368295.1</td>
<td>Chromosome</td>
<td>821153</td>
<td>1820618472</td>
<td>2018-08-09</td>
<td>83650</td>
</tr>
<tr class="odd">
<td>Cyprinus carpio</td>
<td>ASM1834038v1</td>
<td>GCF_018340385.1</td>
<td>Chromosome</td>
<td>1558716</td>
<td>1680118328</td>
<td>2021-05-12</td>
<td>59559</td>
</tr>
<tr class="even">
<td>Danionella translucida</td>
<td>ASM722483v1</td>
<td>GCA_007224835.1</td>
<td>Scaffold</td>
<td>133131</td>
<td>735303417</td>
<td>2019-07-22</td>
<td>35803</td>
</tr>
<tr class="odd">
<td>Danio rerio</td>
<td>GRCz11</td>
<td>GCF_000002035.6</td>
<td>Chromosome</td>
<td>1422317</td>
<td>1373454788</td>
<td>2017-05-09</td>
<td>40031</td>
</tr>
<tr class="even">
<td>Onychostoma macrolepis</td>
<td>ASM1243209v1</td>
<td>GCA_012432095.1</td>
<td>Chromosome</td>
<td>10809623</td>
<td>886569841</td>
<td>2020-04-17</td>
<td>24754</td>
</tr>
<tr class="odd">
<td>Pimephales promelas</td>
<td>EPA_FHM_2.0</td>
<td>GCA_016745375.1</td>
<td>Scaffold</td>
<td>295770</td>
<td>1066412313</td>
<td>2021-01-24</td>
<td>26150</td>
</tr>
<tr class="even">
<td>Puntigrus tetrazona</td>
<td>ASM1883169v1</td>
<td>GCF_018831695.1</td>
<td>Chromosome</td>
<td>1423374</td>
<td>730801986</td>
<td>2021-06-10</td>
<td>40303</td>
</tr>
<tr class="odd">
<td>Sinocyclocheilus anshuiensis</td>
<td>SAMN03320099.WGS_v1.1</td>
<td>GCF_001515605.1</td>
<td>Scaffold</td>
<td>17271</td>
<td>1632701648</td>
<td>2015-12-14</td>
<td>52005</td>
</tr>
<tr class="even">
<td>Sinocyclocheilus grahami</td>
<td>SAMN03320097.WGS_v1.1</td>
<td>GCF_001515645.1</td>
<td>Scaffold</td>
<td>29354</td>
<td>1750271176</td>
<td>2015-12-16</td>
<td>55200</td>
</tr>
<tr class="odd">
<td>Sinocyclocheilus rhinocerous</td>
<td>SAMN03320098_v1.1</td>
<td>GCF_001515625.1</td>
<td>Scaffold</td>
<td>18758</td>
<td>1655769822</td>
<td>2015-12-14</td>
<td>53875</td>
</tr>
<tr class="even">
<td>Triplophysa tibetana</td>
<td>ASM836982v1</td>
<td>GCA_008369825.1</td>
<td>Chromosome</td>
<td>2833080</td>
<td>652930317</td>
<td>2019-09-12</td>
<td>24398</td>
</tr>
</tbody>
</table>
</div>
<h3 id="ddrad-seq-snp-discovery">ddRAD-Seq &amp; SNP Discovery</h3>
<h4 id="worldfish-fin-clipping-collection">WORLDFISH – Fin clipping collection</h4>
<h4 id="zenaida-ddrad-seq-method">ZENAIDA – ddRAD-Seq method</h4>
<h4 id="corrinne-snp-discovery-population-analyses-interested-in-using-the-gender-which-contigs-associated-with-sex">CORRINNE – SNP Discovery, population analyses, interested in using the gender, which contigs associated with sex</h4>
<h3 id="data-availability">Data Availability</h3>
<p>The data used for the <em>L. rohita</em> genome and annotation are available at NCBI
under the BioProject PRJNA650519. The assembled genome sequence and annotations
are available at GenBank under accessions JACTAM000000000. The raw data is
available at the SRA (Sequence Read Archive) under accessions SRR12580210 –
SRR12580221.</p>
<h2 id="results-discussion">Results &amp; Discussion</h2>
<!-- 
Results and Discussion: Should include a description of the genomes, coverage 
amount of genome sequenced, details of sequence variants ascertained, 
identifications of deletions, comparison of strains/genomes (if reporting WGS 
of multiple strains/organisms), and relevance of findings to overall 
organism/strain catalog.
-->
<h3 id="sequencing-assembly">Sequencing &amp; Assembly</h3>
<p>The C-value of <em>L. rohita</em> was previously reported as 1.99 pg (~1.95Gb) using Feulgen densitometry <span class="citation" data-cites="2k3nqSpe">[<a href="#ref-2k3nqSpe" role="doc-biblioref">34</a>]</span> or 1.5Gb using k-mer estimation <span class="citation" data-cites="1EfotvF0W">[<a href="#ref-1EfotvF0W" role="doc-biblioref">35</a>]</span>.
However, the flow cytometry results (Table [<a href="#tbl:flowcyto">3</a>]) show a C-value of 0.99 pg (~0.97Gb) with a standard deviation of 0.02 across all measurements.
The smaller C-value is also closer to the genome estimate produced by GenomeScope (0.97Gb) and the final genome assembly size of 0.95 Gb.</p>
<div id="tbl:flowcyto" class="tablenos">
<table id="tbl:flowcyto" style="width:100%;">
<caption><span>Table 3:</span> Flow cytometry results for 5 <em>L. rohita</em> blood samples, measured twice.
1) Trout erythrocyte nuclei: Genome size = 5.19pg.
2) Genome estimate calculated as (average sample fluorescence/ average standard fluorescence * standard genome size) in picograms DIPLOID (i.e., 2C).
</caption>
<colgroup>
<col style="width: 10%" />
<col style="width: 14%" />
<col style="width: 16%" />
<col style="width: 15%" />
<col style="width: 17%" />
<col style="width: 15%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="header">
<th>Specimen Name</th>
<th>Number of Sample nuclei</th>
<th>Average sample fluorescence</th>
<th>Number of standard nuclei</th>
<th>Average standard fluorescence</th>
<th>Estimated Genome size <sup>2</sup></th>
<th>HAPLOID</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fish 1 Sample 1</td>
<td>16020</td>
<td>27350</td>
<td>2065</td>
<td>69247</td>
<td>2.049857756</td>
<td>1.024928878</td>
</tr>
<tr class="even">
<td>Fish 2 Sample 1</td>
<td>13082</td>
<td>25929</td>
<td>6570</td>
<td>66671</td>
<td>2.018441451</td>
<td>1.0092207255</td>
</tr>
<tr class="odd">
<td>Fish 2 Sample 2</td>
<td>15402</td>
<td>25665</td>
<td>4354</td>
<td>67489</td>
<td>1.973674969</td>
<td>0.9868374845</td>
</tr>
<tr class="even">
<td>Fish 3 Sample 1</td>
<td>15124</td>
<td>25798</td>
<td>4442</td>
<td>68195</td>
<td>1.963364176</td>
<td>0.981682088</td>
</tr>
<tr class="odd">
<td>Fish 3 Sample 2</td>
<td>14923</td>
<td>25763</td>
<td>4823</td>
<td>68837</td>
<td>1.942414254</td>
<td>0.971207127</td>
</tr>
<tr class="even">
<td>Fish 4 Sample 1</td>
<td>13320</td>
<td>26346</td>
<td>5913</td>
<td>69665</td>
<td>1.962760927</td>
<td>0.9813804635</td>
</tr>
<tr class="odd">
<td>Fish 4 Sample 2</td>
<td>5624</td>
<td>26612</td>
<td>4097</td>
<td>68876</td>
<td>2.005288925</td>
<td>1.0026444625</td>
</tr>
<tr class="even">
<td>Fish 5 Sample 1</td>
<td>6771</td>
<td>25761</td>
<td>3080</td>
<td>68825</td>
<td>1.942602107</td>
<td>0.9713010535</td>
</tr>
<tr class="odd">
<td>Fish 5 Sample 2</td>
<td>15926</td>
<td>26369</td>
<td>3352</td>
<td>68832</td>
<td>1.988248344</td>
<td>0.994124172</td>
</tr>
<tr class="even">
<td>Standard only <sup>1</sup></td>
<td>3</td>
<td>25258</td>
<td>3311</td>
<td>64331</td>
<td>NA</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td></td>
<td>Average</td>
<td>1.982961434</td>
<td>0.9914807172</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td>Standard Deviation</td>
<td>0.03607582</td>
<td>0.01803790999</td>
</tr>
</tbody>
</table>
</div>
<p>A total of 130.5 Gb of Nanopore long reads from 44.7 million read, and 261 Gb of Illumina short reads from 870 million pairs were produced, along with 382 million pairs (114 Gb) for the Hi-C library.
The initial <em>de novo</em> assembly consisted of 4999 contigs with an N50 of 1.28 Mb.
After the Bionano and HiC data was incorporated, the total number of sequences dropped to 2899 and the N50 increased to 29.9 Mb.
The final assembly consisted of 25 chromosome length scaffolds and 2844 unplaced scaffolds, ranging in size from 1,479bp to 7.18 Mb.
The chromosome scaffold consist of one to eight sequences with all but three made from three or fewer sequences.
Table [<a href="#tbl:assembly">4</a>] contains a common assembly statistics for each step.
The final genome size is 97.9% of the estimated genome size.
The annotation pipeline produced 51,079 primary transcripts, 31,274 surviving the AED, gap, and overlapping filters.
BUSCO analysis show the genome completely contains 98.1% of the 3640 orthologs in the actinopterygii_odb10 database with 37 (1%) duplicated;
while, the filtered transcriptome contains 84.5% of the total orthologs complete with 74 duplicated.
A complete comparison of the BUSCO analyses can be found in Table [<a href="#tbl:busco">5</a>].</p>
<div id="tbl:assembly" class="tablenos">
<table id="tbl:assembly">
<caption><span>Table 4:</span> Assembly statistics for each stage of the assembly
</caption>
<thead>
<tr class="header">
<th>n</th>
<th>n:500</th>
<th>L50</th>
<th>min</th>
<th>N75</th>
<th>N50</th>
<th>N25</th>
<th>E-size</th>
<th>max</th>
<th>sum</th>
<th>name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4999</td>
<td>4999</td>
<td>202</td>
<td>1348</td>
<td>514919</td>
<td>1281850</td>
<td>2395030</td>
<td>1727184</td>
<td>7832582</td>
<td>9.43E+08</td>
<td>wtdbg2</td>
</tr>
<tr class="even">
<td>3709</td>
<td>3706</td>
<td>15</td>
<td>1479</td>
<td>1.13E+07</td>
<td>2.65E+07</td>
<td>3.08E+07</td>
<td>2.20E+07</td>
<td>3.79E+07</td>
<td>9.46E+08</td>
<td>bionano</td>
</tr>
<tr class="odd">
<td>2899</td>
<td>2896</td>
<td>14</td>
<td>1479</td>
<td>2.64E+07</td>
<td>2.99E+07</td>
<td>3.43E+07</td>
<td>2.69E+07</td>
<td>4.45E+07</td>
<td>9.46E+08</td>
<td>hic</td>
</tr>
<tr class="even">
<td>2872</td>
<td>2869</td>
<td>13</td>
<td>1479</td>
<td>2.88E+07</td>
<td>3.25E+07</td>
<td>3.61E+07</td>
<td>3.00E+07</td>
<td>4.53E+07</td>
<td>9.46E+08</td>
<td>ragtag</td>
</tr>
</tbody>
</table>
</div>
<div id="tbl:busco" class="tablenos">
<table id="tbl:busco">
<caption><span>Table 5:</span> BUSCO analysis for the genome and transcriptome, before and after AED filtering.
</caption>
<thead>
<tr class="header">
<th>Type</th>
<th>Genome</th>
<th>Unfiltered Transcriptome</th>
<th>Filtered Transcriptome</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Complete BUSCOs (C)</td>
<td>3571</td>
<td>3139</td>
<td>3078</td>
</tr>
<tr class="even">
<td>Complete and single-copy BUSCOs (S)</td>
<td>3534</td>
<td>3064</td>
<td>3001</td>
</tr>
<tr class="odd">
<td>Complete and duplicated BUSCOs (D)</td>
<td>37</td>
<td>75</td>
<td>74</td>
</tr>
<tr class="even">
<td>Fragmented BUSCOs (F)</td>
<td>23</td>
<td>192</td>
<td>170</td>
</tr>
<tr class="odd">
<td>Missing BUSCOs (M)</td>
<td>46</td>
<td>309</td>
<td>392</td>
</tr>
<tr class="even">
<td>Total BUSCO groups searched</td>
<td>3640</td>
<td>3640</td>
<td>3640</td>
</tr>
</tbody>
</table>
</div>
<pre><code>================================================== 
file name: Rohu.genome.fa            
sequences:          2872 
total length: 1128029156 bp (945637473 bp excl N/X-runs) 
GC level:         36.05 % 
bases masked:  465289941 bp (41.25 %) 
================================================== 
               number of      length   percentage 
               elements*    occupied  of sequence 
-------------------------------------------------- 
Retroelements       123482     53424977 bp    4.74 % 
   SINEs:                0            0 bp    0.00 % 
   Penelope              0            0 bp    0.00 % 
   LINEs:            29838     13842585 bp    1.23 % 
    CRE/SLACS            0            0 bp    0.00 % 
     L2/CR1/Rex      15537      8265099 bp    0.73 % 
     R1/LOA/Jockey     981       643697 bp    0.06 % 
     R2/R4/NeSL        936       732043 bp    0.06 % 
     RTE/Bov-B         271       181676 bp    0.02 % 
     L1/CIN4          7978      1801119 bp    0.16 % 
   LTR elements:     93644     39582392 bp    3.51 % 
     BEL/Pao          1287      1138420 bp    0.10 % 
     Ty1/Copia         221       110739 bp    0.01 % 
     Gypsy/DIRS1     32714     19073624 bp    1.69 % 
       Retroviral     2443      1815904 bp    0.16 % 
  
DNA transposons      79815     22485920 bp    1.99 % 
   hobo-Activator     2242       639067 bp    0.06 % 
   Tc1-IS630-Pogo    49209     16661924 bp    1.48 % 
   En-Spm                0            0 bp    0.00 % 
   MuDR-IS905            0            0 bp    0.00 % 
   PiggyBac              0            0 bp    0.00 % 
   Tourist/Harbinger  1794       259934 bp    0.02 % 
   Other (Mirage,      900       209002 bp    0.02 % 
    P-element, Transib) 
  
Rolling-circles        196        71055 bp    0.01 % 
  
Unclassified:       1774260    361869982 bp   32.08 % 
  
Total interspersed repeats:   437780879 bp   38.81 % 
  
  
Small RNA:               0            0 bp    0.00 % 
  
Satellites:              1          267 bp    0.00 % 
Simple repeats:     504151     23839806 bp    2.11 % 
Low complexity:      58354      3597934 bp    0.32 % 
================================================== 
  
* most repeats fragmented by insertions or deletions 
  have been counted as one element 
                                                       
  
RepeatMasker version 4.1.1 , default mode 
                                         
run with rmblastn version 2.10.0+ 
The query was compared to classified sequences in &quot;Rohu-families.fa&quot; 
 </code></pre>
<h3 id="comparative-genomics">Comparative genomics</h3>
<h4 id="labeo-rohita">Labeo Rohita</h4>
<div id="tbl:assembly-stats" class="tablenos">
<table id="tbl:assembly-stats">
<caption><span>Table 6:</span> Assembly stats for rohu genomes
</caption>
<thead>
<tr class="header">
<th>Name</th>
<th>n</th>
<th>n:500</th>
<th>L50</th>
<th>min</th>
<th>N75</th>
<th>N50</th>
<th>N25</th>
<th>E-size</th>
<th>max</th>
<th>sum</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CIFA</td>
<td>13623</td>
<td>13617</td>
<td>182</td>
<td>500</td>
<td>774698</td>
<td>2006958</td>
<td>4279019</td>
<td>2905260</td>
<td>15.15e6</td>
<td>1.427e9</td>
</tr>
<tr class="even">
<td>HRRL</td>
<td>13660</td>
<td>13650</td>
<td>182</td>
<td>500</td>
<td>774698</td>
<td>2006958</td>
<td>4279019</td>
<td>2905216</td>
<td>15.15e6</td>
<td>1.427e9</td>
</tr>
<tr class="odd">
<td>IGBB</td>
<td>2869</td>
<td>2869</td>
<td>13</td>
<td>1479</td>
<td>28.79e6</td>
<td>32.5e6</td>
<td>36.06e6</td>
<td>30.03e6</td>
<td>45.29e6</td>
<td>945.5e6</td>
</tr>
</tbody>
</table>
</div>
<p><img src="../comp-gen/1-stats/length.rohu.hires.png" title="Length distro" alt="Length distribution of scaffolds for each rohu assembly. Each bar represents the number of bases contained in scaffolds of a particular length." />[#fig:len-distro]</p>
<p><img src="../comp-gen/1-busco/rohu/busco.summary.png" title="Rohu BUSCOs" alt="BUSCO score for each rohu genome" />[#fig:busco-rohu]</p>
<p><img src="../comp-gen/2-dotplot/rohu.png" alt="Dot-plots comparing sequence similarity between rohu genomes" />[#fig:dotplot-rohu]</p>
<h4 id="cypriniformes">Cypriniformes</h4>
<p><img src="../comp-gen/1-busco/cypriniformes/busco.summary.png" title="Rohu BUSCOs" alt="BUSCO score for each rohu genome" />[#fig:busco-carps]</p>
<p><img src="../comp-gen/2-dotplot/cypriniformes.rm-uninformative.png" alt="Dot-plots comparing sequence similarity between rohu genomes" />[#fig:dotplot-carps]</p>
<div id="tbl:orthofinder" class="tablenos">
<table id="tbl:orthofinder">
<caption><span>Table 7:</span> Summary statistics per species from OrthoFinder.
</caption>
<colgroup>
<col style="width: 15%" />
<col style="width: 9%" />
<col style="width: 11%" />
<col style="width: 9%" />
<col style="width: 17%" />
<col style="width: 15%" />
<col style="width: 20%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>Number of Genes</th>
<th>Genes in Orthogroups</th>
<th>Unassigned Genes</th>
<th>Orthogroups Containing Species</th>
<th>Species-specific Orthogroups</th>
<th>Genes in Species-specific Orthogroups</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Anabarilius grahami</td>
<td>23906</td>
<td>21942 (91.8%)</td>
<td>1964 (8.2%)</td>
<td>15217 (51.5%)</td>
<td>180</td>
<td>888 (3.7%)</td>
</tr>
<tr class="even">
<td>Carassius auratus</td>
<td>96703</td>
<td>93988 (97.2%)</td>
<td>2715 (2.8%)</td>
<td>21896 (74.2%)</td>
<td>472</td>
<td>1775 (1.8%)</td>
</tr>
<tr class="odd">
<td>Cyprinus carpio</td>
<td>80686</td>
<td>78684 (97.5%)</td>
<td>2002 (2.5%)</td>
<td>21584 (73.1%)</td>
<td>287</td>
<td>964 (1.2%)</td>
</tr>
<tr class="even">
<td>Danio rerio</td>
<td>52829</td>
<td>51951 (98.3%)</td>
<td>878 (1.7%)</td>
<td>20671 (70.0%)</td>
<td>392</td>
<td>2184 (4.1%)</td>
</tr>
<tr class="odd">
<td>Danionella translucida</td>
<td>35381</td>
<td>32943 (93.1%)</td>
<td>2438 (6.9%)</td>
<td>19073 (64.6%)</td>
<td>540</td>
<td>1941 (5.5%)</td>
</tr>
<tr class="even">
<td>Labeo rohita</td>
<td>31274</td>
<td>29904 (95.6%)</td>
<td>1370 (4.4%)</td>
<td>18740 (63.5%)</td>
<td>161</td>
<td>1581 (5.1%)</td>
</tr>
<tr class="odd">
<td>Onychostoma macrolepis</td>
<td>24754</td>
<td>24483 (98.9%)</td>
<td>271 (1.1%)</td>
<td>19276 (65.3%)</td>
<td>137</td>
<td>603 (2.4%)</td>
</tr>
<tr class="even">
<td>Pimephales promelas</td>
<td>47578</td>
<td>45412 (95.4%)</td>
<td>2166 (4.6%)</td>
<td>19884 (67.4%)</td>
<td>506</td>
<td>1826 (3.8%)</td>
</tr>
<tr class="odd">
<td>Puntigrus tetrazona</td>
<td>48681</td>
<td>48094 (98.8%)</td>
<td>587 (1.2%)</td>
<td>20582 (69.7%)</td>
<td>129</td>
<td>517 (1.1%)</td>
</tr>
<tr class="even">
<td>Sinocyclocheilus anshuiensis</td>
<td>68474</td>
<td>66456 (97.1%)</td>
<td>2018 (2.9%)</td>
<td>21485 (72.8%)</td>
<td>114</td>
<td>344 (0.5%)</td>
</tr>
<tr class="odd">
<td>Sinocyclocheilus grahami</td>
<td>67410</td>
<td>63316 (93.9%)</td>
<td>4094 (6.1%)</td>
<td>22326 (75.6%)</td>
<td>338</td>
<td>793 (1.2%)</td>
</tr>
<tr class="even">
<td>Sinocyclocheilus rhinocerous</td>
<td>68562</td>
<td>65831 (96.0%)</td>
<td>2731 (4.0%)</td>
<td>21884 (74.1%)</td>
<td>172</td>
<td>414 (0.6%)</td>
</tr>
<tr class="odd">
<td>Triplophysa tibetana</td>
<td>24310</td>
<td>23279 (95.8%)</td>
<td>1031 (4.2%)</td>
<td>18734 (63.5%)</td>
<td>125</td>
<td>480 (2.0%)</td>
</tr>
</tbody>
</table>
</div>
<p><em>SNP discovery and population similarities among rohu fisheries</em></p>
<p><em>Labeo rohita</em> is an important aquaculture fish in many areas of south Asia [I
don’t know how to put in this reference FAO (2019) Food and Agricultural
Organization. Yearbook of Fishery and Aquaculture Statistics 2017].</p>
<p>The high fecundity and relative ease of cultivation has made rohu successful in
breeding programs, leading to the generation of a fast-growing strain called
Jayanti Historically, rohu breeding and research has relied on low-throughput
information, such as cloning individual genes of interest (reviewed in
<span class="citation" data-cites="1Hd37EDan">[<a href="#ref-1Hd37EDan" role="doc-biblioref">36</a>]</span>) or microsatellite-based selective breeding
<span class="citation" data-cites="USLcrYsW">[<a href="#ref-USLcrYsW" role="doc-biblioref">37</a>]</span>; however, recent years have seen increased
interest in applying high-throughput methods to rohu breeding, such as SNP
panels <span class="citation" data-cites="U0nH39nP">[<a href="#ref-U0nH39nP" role="doc-biblioref">38</a>]</span> <span class="citation" data-cites="19VelQ7B6">[<a href="#ref-19VelQ7B6" role="doc-biblioref">39</a>]</span>,
genotyping-by-sequencing <span class="citation" data-cites="640Eq04I">[<a href="#ref-640Eq04I" role="doc-biblioref">40</a>]</span>, transcriptomics
<span class="citation" data-cites="U0nH39nP">[<a href="#ref-U0nH39nP" role="doc-biblioref">38</a>]</span> <span class="citation" data-cites="RRMEOwZ6">[<a href="#ref-RRMEOwZ6" role="doc-biblioref">41</a>]</span>, and
degradomics <span class="citation" data-cites="HUT5rWB4">[<a href="#ref-HUT5rWB4" role="doc-biblioref">42</a>]</span>. Recently, a draft genome was
published for L. rohita along with pooled resequencing from three individuals in
10 populations <span class="citation" data-cites="hpPKbDbP">[<a href="#ref-hpPKbDbP" role="doc-biblioref">43</a>]</span>; however, genetic diversity and
divergence among the populations was not reported.</p>
<h2 id="acknowledgements">Acknowledgements</h2>
<p>ISU Office of Biotechnology Flow Cytometry Facility</p>
<p>ResearchIT Iowa State University</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-11lZmTCLG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>The application of flow cytometry for estimating genome size and ploidy level in plants.</strong> <div class="csl-block">Jaume Pellicer, Ilia J Leitch</div> <em>Methods in molecular biology (Clifton, N.J.)</em> (2014) <a href="https://www.ncbi.nlm.nih.gov/pubmed/24415480">https://www.ncbi.nlm.nih.gov/pubmed/24415480</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-1-62703-767-9_14">10.1007/978-1-62703-767-9_14</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24415480">24415480</a></div></div>
</div>
<div id="ref-14dfsRv4c" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>A fast, lock-free approach for efficient parallel counting of occurrences of k-mers</strong> <div class="csl-block">Guillaume Marçais, Carl Kingsford</div> <em>Bioinformatics</em> (2011-03-15) <a href="https://doi.org/b7gkd6">https://doi.org/b7gkd6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btr011">10.1093/bioinformatics/btr011</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21217122">21217122</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3051319">PMC3051319</a></div></div>
</div>
<div id="ref-1H1zfojOz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>GenomeScope: fast reference-free genome profiling from short reads</strong> <div class="csl-block">Gregory W Vurture, Fritz J Sedlazeck, Maria Nattestad, Charles J Underwood, Han Fang, James Gurtowski, Michael C Schatz</div> <em>Bioinformatics (Oxford, England)</em> (2017-07-15) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5870704/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5870704/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btx153">10.1093/bioinformatics/btx153</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28369201">28369201</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5870704">PMC5870704</a></div></div>
</div>
<div id="ref-88NBy0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>NanoPack: visualizing and processing long-read sequencing data</strong> <div class="csl-block">Wouter De Coster, Svenn D’Hert, Darrin T Schultz, Marc Cruts, Christine Van Broeckhoven</div> <em>Bioinformatics</em> (2018-08-01) <a href="https://doi.org/gf38cx">https://doi.org/gf38cx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/bty149">10.1093/bioinformatics/bty149</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29547981">29547981</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6061794">PMC6061794</a></div></div>
</div>
<div id="ref-l4tjRGIB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Trimmomatic: a flexible trimmer for Illumina sequence data</strong> <div class="csl-block">Anthony M Bolger, Marc Lohse, Bjoern Usadel</div> <em>Bioinformatics</em> (2014-08-01) <a href="https://doi.org/f6cj5w">https://doi.org/f6cj5w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btu170">10.1093/bioinformatics/btu170</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24695404">24695404</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590">PMC4103590</a></div></div>
</div>
<div id="ref-9l1p27cN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Fast and accurate long-read assembly with wtdbg2</strong> <div class="csl-block">Jue Ruan, Heng Li</div> <em>Nature Methods</em> (2020-02) <a href="https://doi.org/ggd8j9">https://doi.org/ggd8j9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-019-0669-3">10.1038/s41592-019-0669-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31819265">31819265</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7004874">PMC7004874</a></div></div>
</div>
<div id="ref-zY4KP2bZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Fast and accurate de novo genome assembly from long uncorrected reads</strong> <div class="csl-block">Robert Vaser, Ivan Sović, Niranjan Nagarajan, Mile Šikić</div> <em>Genome Research</em> (2017-05) <a href="https://doi.org/gfkmzr">https://doi.org/gfkmzr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/gr.214270.116">10.1101/gr.214270.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28100585">28100585</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411768">PMC5411768</a></div></div>
</div>
<div id="ref-172cJaw4Q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Minimap2: pairwise alignment for nucleotide sequences</strong> <div class="csl-block">Heng Li</div> <em>Bioinformatics</em> (2018-09-15) <a href="https://doi.org/gdhbqt">https://doi.org/gdhbqt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29750242">29750242</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6137996">PMC6137996</a></div></div>
</div>
<div id="ref-LDP9ZoBM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</strong> <div class="csl-block">Bruce J Walker, Thomas Abeel, Terrance Shea, Margaret Priest, Amr Abouelliel, Sharadha Sakthikumar, Christina A Cuomo, Qiandong Zeng, Jennifer Wortman, Sarah K Young, Ashlee M Earl</div> <em>PLoS ONE</em> (2014-11-19) <a href="https://doi.org/gfkmz5">https://doi.org/gfkmz5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0112963">10.1371/journal.pone.0112963</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25409509">25409509</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4237348">PMC4237348</a></div></div>
</div>
<div id="ref-roi5bwXL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Fast and accurate short read alignment with Burrows-Wheeler transform</strong> <div class="csl-block">H Li, R Durbin</div> <em>Bioinformatics</em> (2009-07-15) <a href="https://doi.org/dqt59j">https://doi.org/dqt59j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btp324">10.1093/bioinformatics/btp324</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19451168">19451168</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2705234">PMC2705234</a></div></div>
</div>
<div id="ref-wgWE6D4o" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Bionano Access Software</strong> <div class="csl-block">Bionano Genomics</div> <a href="https://bionanogenomics.com/support-page/bionano-access-software/">https://bionanogenomics.com/support-page/bionano-access-software/</a></div>
</div>
<div id="ref-KEqlQin" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Integrating Hi-C links with assembly graphs for chromosome-scale assembly</strong> <div class="csl-block">Jay Ghurye, Arang Rhie, Brian P Walenz, Anthony Schmitt, Siddarth Selvaraj, Mihai Pop, Adam M Phillippy, Sergey Koren</div> <em>PLOS Computational Biology</em> (2019-08-21) <a href="https://doi.org/gf74qw">https://doi.org/gf74qw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1007273">10.1371/journal.pcbi.1007273</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31433799">31433799</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6719893">PMC6719893</a></div></div>
</div>
<div id="ref-GNj8xY3t" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>Chromosome‐level genome assembly of a cyprinid fish &lt;i&gt;Onychostoma macrolepis&lt;/i&gt; by integration of nanopore sequencing, Bionano and Hi‐C technology</strong> <div class="csl-block">Lina Sun, Tian Gao, Feilong Wang, Zuliang Qin, Longxia Yan, Wenjing Tao, Minghui Li, Canbiao Jin, Li Ma, Thomas D Kocher, Deshou Wang</div> <em>Molecular Ecology Resources</em> (2020-09) <a href="https://doi.org/gmx33w">https://doi.org/gmx33w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/1755-0998.13190">10.1111/1755-0998.13190</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32419357">32419357</a></div></div>
</div>
<div id="ref-zePiKkRw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>RaGOO: fast and accurate reference-guided scaffolding of draft genomes</strong> <div class="csl-block">Michael Alonge, Sebastian Soyk, Srividya Ramakrishnan, Xingang Wang, Sara Goodwin, Fritz J Sedlazeck, Zachary B Lippman, Michael C Schatz</div> <em>Genome Biology</em> (2019-12) <a href="https://doi.org/ggctnf">https://doi.org/ggctnf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-019-1829-6">10.1186/s13059-019-1829-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31661016">31661016</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6816165">PMC6816165</a></div></div>
</div>
<div id="ref-DUW8OYnu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>RepeatModeler2 for automated genomic discovery of transposable element families</strong> <div class="csl-block">Jullien M Flynn, Robert Hubley, Clément Goubert, Jeb Rosen, Andrew G Clark, Cédric Feschotte, Arian F Smit</div> <em>Proceedings of the National Academy of Sciences</em> (2020-04-28) <a href="https://doi.org/ggsnv2">https://doi.org/ggsnv2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.1921046117">10.1073/pnas.1921046117</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32300014">32300014</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7196820">PMC7196820</a></div></div>
</div>
<div id="ref-1GLBRcU8y" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Graph-based genome alignment and genotyping with HISAT2 and HISAT-genotype</strong> <div class="csl-block">Daehwan Kim, Joseph M Paggi, Chanhee Park, Christopher Bennett, Steven L Salzberg</div> <em>Nature Biotechnology</em> (2019-08) <a href="https://doi.org/gf5395">https://doi.org/gf5395</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41587-019-0201-4">10.1038/s41587-019-0201-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31375807">31375807</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7605509">PMC7605509</a></div></div>
</div>
<div id="ref-Nc6oNbiA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Leveraging multiple transcriptome assembly methods for improved gene structure annotation</strong> <div class="csl-block">Luca Venturini, Shabhonam Caim, Gemy George Kaithakottil, Daniel Lee Mapleson, David Swarbreck</div> <em>GigaScience</em> (2018-08-01) <a href="https://doi.org/gmx7k4">https://doi.org/gmx7k4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giy093">10.1093/gigascience/giy093</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30052957">30052957</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6105091">PMC6105091</a></div></div>
</div>
<div id="ref-Sq0xBB4s" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>BRAKER2: automatic eukaryotic genome annotation with GeneMark-EP+ and AUGUSTUS supported by a protein database</strong> <div class="csl-block">Tomáš Brůna, Katharina J Hoff, Alexandre Lomsadze, Mario Stanke, Mark Borodovsky</div> <em>NAR Genomics and Bioinformatics</em> (2021-01-06) <a href="https://doi.org/gmx7k6">https://doi.org/gmx7k6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nargab/lqaa108">10.1093/nargab/lqaa108</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33575650">33575650</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7787252">PMC7787252</a></div></div>
</div>
<div id="ref-1F8s8oDfw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>Transcriptome assembly from long-read RNA-seq alignments with StringTie2</strong> <div class="csl-block">Sam Kovaka, Aleksey V Zimin, Geo M Pertea, Roham Razaghi, Steven L Salzberg, Mihaela Pertea</div> <em>Genome Biology</em> (2019-12) <a href="https://doi.org/ghrk2k">https://doi.org/ghrk2k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-019-1910-1">10.1186/s13059-019-1910-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31842956">31842956</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6912988">PMC6912988</a></div></div>
</div>
<div id="ref-14kgcZI3s" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks</strong> <div class="csl-block">Cole Trapnell, Adam Roberts, Loyal Goff, Geo Pertea, Daehwan Kim, David R Kelley, Harold Pimentel, Steven L Salzberg, John L Rinn, Lior Pachter</div> <em>Nature Protocols</em> (2012-03) <a href="https://doi.org/f4pbzd">https://doi.org/f4pbzd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nprot.2012.016">10.1038/nprot.2012.016</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22383036">22383036</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3334321">PMC3334321</a></div></div>
</div>
<div id="ref-oU9Lbgmd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>Full-length transcriptome assembly from RNA-Seq data without a reference genome</strong> <div class="csl-block">Manfred G Grabherr, Brian J Haas, Moran Yassour, Joshua Z Levin, Dawn A Thompson, Ido Amit, Xian Adiconis, Lin Fan, Raktima Raychowdhury, Qiandong Zeng, … Aviv Regev</div> <em>Nature Biotechnology</em> (2011-07) <a href="https://doi.org/b2bctj">https://doi.org/b2bctj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nbt.1883">10.1038/nbt.1883</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21572440">21572440</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3571712">PMC3571712</a></div></div>
</div>
<div id="ref-B9kt43k8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>Efficient and accurate detection of splice junctions from RNA-seq with Portcullis</strong> <div class="csl-block">Daniel Mapleson, Luca Venturini, Gemy Kaithakottil, David Swarbreck</div> <em>GigaScience</em> (2018-12-01) <a href="https://doi.org/gkzg5r">https://doi.org/gkzg5r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giy131">10.1093/gigascience/giy131</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30418570">30418570</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6302956">PMC6302956</a></div></div>
</div>
<div id="ref-lX665mdh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>Prodigal: prokaryotic gene recognition and translation initiation site identification</strong> <div class="csl-block">Doug Hyatt, Gwo-Liang Chen, Philip F LoCascio, Miriam L Land, Frank W Larimer, Loren J Hauser</div> <em>BMC Bioinformatics</em> (2010-12) <a href="https://doi.org/cktxnm">https://doi.org/cktxnm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-11-119">10.1186/1471-2105-11-119</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20211023">20211023</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2848648">PMC2848648</a></div></div>
</div>
<div id="ref-7OpVD07l" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>Eukaryotic Gene Prediction Using GeneMark.hmm‐E and GeneMark‐ES</strong> <div class="csl-block">Mark Borodovsky, Alex Lomsadze</div> <em>Current Protocols in Bioinformatics</em> (2011-09) <a href="https://doi.org/b75r2g">https://doi.org/b75r2g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/0471250953.bi0406s35">10.1002/0471250953.bi0406s35</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21901742">21901742</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3204378">PMC3204378</a></div></div>
</div>
<div id="ref-5gYwalTK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Gene prediction in eukaryotes with a generalized hidden Markov model that uses hints from external sources</strong> <div class="csl-block">Mario Stanke, Oliver Schöffmann, Burkhard Morgenstern, Stephan Waack</div> <em>BMC Bioinformatics</em> (2006-12) <a href="https://doi.org/cv8xsn">https://doi.org/cv8xsn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-7-62">10.1186/1471-2105-7-62</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16469098">16469098</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1409804">PMC1409804</a></div></div>
</div>
<div id="ref-UdSgHXyu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>MAKER2: an annotation pipeline and genome-database management tool for second-generation genome projects</strong> <div class="csl-block">Carson Holt, Mark Yandell</div> <em>BMC Bioinformatics</em> (2011-12) <a href="https://doi.org/fz39nj">https://doi.org/fz39nj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-12-491">10.1186/1471-2105-12-491</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22192575">22192575</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3280279">PMC3280279</a></div></div>
</div>
<div id="ref-N1v57w5y" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>InterProScan 5: genome-scale protein function classification</strong> <div class="csl-block">P Jones, D Binns, H-Y Chang, M Fraser, W Li, C McAnulla, H McWilliam, J Maslen, A Mitchell, G Nuka, … S Hunter</div> <em>Bioinformatics</em> (2014-05-01) <a href="https://doi.org/f53532">https://doi.org/f53532</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btu031">10.1093/bioinformatics/btu031</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24451626">24451626</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3998142">PMC3998142</a></div></div>
</div>
<div id="ref-nEsJGUWa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>BLAST+: architecture and applications</strong> <div class="csl-block">Christiam Camacho, George Coulouris, Vahram Avagyan, Ning Ma, Jason Papadopoulos, Kevin Bealer, Thomas L Madden</div> <em>BMC Bioinformatics</em> (2009-12) <a href="https://doi.org/cnjxgz">https://doi.org/cnjxgz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-10-421">10.1186/1471-2105-10-421</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20003500">20003500</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2803857">PMC2803857</a></div></div>
</div>
<div id="ref-Vq0QVho4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>BUSCO Update: Novel and Streamlined Workflows along with Broader and Deeper Phylogenetic Coverage for Scoring of Eukaryotic, Prokaryotic, and Viral Genomes</strong> <div class="csl-block">Mosè Manni, Matthew R Berkeley, Mathieu Seppey, Felipe A Simão, Evgeny M Zdobnov</div> <em>Molecular Biology and Evolution</em> (2021-09-27) <a href="https://doi.org/gmgv52">https://doi.org/gmgv52</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/molbev/msab199">10.1093/molbev/msab199</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34320186">34320186</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8476166">PMC8476166</a></div></div>
</div>
<div id="ref-2W9THYfa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>ABySS 2.0: resource-efficient assembly of large genomes using a Bloom filter</strong> <div class="csl-block">Shaun D Jackman, Benjamin P Vandervalk, Hamid Mohamadi, Justin Chu, Sarah Yeo, SAustin Hammond, Golnaz Jahesh, Hamza Khan, Lauren Coombe, Rene L Warren, Inanc Birol</div> <em>Genome Research</em> (2017-05) <a href="https://doi.org/f9x8qp">https://doi.org/f9x8qp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/gr.214346.116">10.1101/gr.214346.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28232478">28232478</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411771">PMC5411771</a></div></div>
</div>
<div id="ref-eyOvxHVn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>Twelve years of SAMtools and BCFtools</strong> <div class="csl-block">Petr Danecek, James K Bonfield, Jennifer Liddle, John Marshall, Valeriu Ohan, Martin O Pollard, Andrew Whitwham, Thomas Keane, Shane A McCarthy, Robert M Davies, Heng Li</div> <em>GigaScience</em> (2021-01-29) <a href="https://doi.org/gjxzc9">https://doi.org/gjxzc9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giab008">10.1093/gigascience/giab008</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33590861">33590861</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7931819">PMC7931819</a></div></div>
</div>
<div id="ref-EWll2pyF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>Welcome to the Tidyverse</strong> <div class="csl-block">Hadley Wickham, Mara Averick, Jennifer Bryan, Winston Chang, Lucy McGowan, Romain François, Garrett Grolemund, Alex Hayes, Lionel Henry, Jim Hester, … Hiroaki Yutani</div> <em>Journal of Open Source Software</em> (2019-11-21) <a href="https://doi.org/ggddkj">https://doi.org/ggddkj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.21105/joss.01686">10.21105/joss.01686</a></div></div>
</div>
<div id="ref-1E6awuyBZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>OrthoFinder: phylogenetic orthology inference for comparative genomics</strong> <div class="csl-block">David M Emms, Steven Kelly</div> <em>Genome Biology</em> (2019-12) <a href="https://doi.org/ggczpr">https://doi.org/ggczpr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-019-1832-y">10.1186/s13059-019-1832-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31727128">31727128</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6857279">PMC6857279</a></div></div>
</div>
<div id="ref-2k3nqSpe" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><a href="http://zotero.org/users/local/gVY292am/items/GYK6YE2I">http://zotero.org/users/local/gVY292am/items/GYK6YE2I</a></div>
</div>
<div id="ref-1EfotvF0W" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>&lt;i&gt;De novo&lt;/i&gt; Assembly and Genome-Wide SNP Discovery in Rohu Carp, &lt;i&gt;Labeo rohita&lt;/i&gt;</strong> <div class="csl-block">Paramananda Das, Lakshman Sahoo, Sofia P Das, Amrita Bit, Chaitanya G Joshi, Basdeo Kushwaha, Dinesh Kumar, Tejas M Shah, Ankit T Hinsu, Namrata Patel, … Joykrushna Jena</div> <em>Frontiers in genetics</em> (2020-04-21) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7186481/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7186481/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fgene.2020.00386">10.3389/fgene.2020.00386</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32373166">32373166</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7186481">PMC7186481</a></div></div>
</div>
<div id="ref-1Hd37EDan" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>Status of genetic and genomic approaches for delineating biological information and improving aquaculture production of farmed rohu, &lt;i&gt;Labeo rohita&lt;/i&gt; (Ham, 1822)</strong> <div class="csl-block">Kiran D Rasal, Jitendra Kumar Sundaray</div> <em>Reviews in Aquaculture</em> (2020-11) <a href="https://doi.org/gnj5hk">https://doi.org/gnj5hk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/raq.12444">10.1111/raq.12444</a></div></div>
</div>
<div id="ref-USLcrYsW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>A molecular tool for parentage analysis in Indian major carp, Labeo rohita (Hamilton, 1822)</strong> <div class="csl-block">L Sahoo, PK Meher, K Das Mahapatra, JN Saha, P Jayasankar, P Das</div> <em>Aquaculture International</em> (2017-06) <a href="https://doi.org/gbgqt5">https://doi.org/gbgqt5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10499-016-0104-z">10.1007/s10499-016-0104-z</a></div></div>
</div>
<div id="ref-U0nH39nP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>A linkage map of transcribed single nucleotide polymorphisms in rohu (Labeo rohita) and QTL associated with resistance to Aeromonas hydrophila</strong> <div class="csl-block">Nicholas Robinson, Matthew Baranski, Kanta Mahapatra, Jatindra Saha, Sweta Das, Jashobanta Mishra, Paramananda Das, Matthew Kent, Mariann Arnyasi, Pramoda Sahoo</div> <em>BMC Genomics</em> (2014) <a href="https://doi.org/gb3g4v">https://doi.org/gb3g4v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2164-15-541">10.1186/1471-2164-15-541</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24984705">24984705</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4226992">PMC4226992</a></div></div>
</div>
<div id="ref-19VelQ7B6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>Status and future perspectives of single nucleotide polymorphisms (SNPs) markers in farmed fishes: Way ahead using next generation sequencing</strong> <div class="csl-block">Kiran Dashrath Rasal, Vemlawada Chakrapani, Amrendra Kumar Pandey, Avinash Rambhau Rasal, Jitendra K Sundaray, Arun Ninawe, Pallipuram Jayasankar</div> <em>Gene Reports</em> (2017-03) <a href="https://doi.org/gjg4d7">https://doi.org/gjg4d7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.genrep.2016.12.004">10.1016/j.genrep.2016.12.004</a></div></div>
</div>
<div id="ref-640Eq04I" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>Single Nucleotide Polymorphisms (SNPs) Reveal Sibship Among Founders of a Bangladeshi Rohu (Labeo rohita) Breeding Population</strong> <div class="csl-block">Matthew Gray Hamilton, Wagdy Mekkawy, Andrzej Kilian, John AH Benzie</div> <em>Frontiers in Genetics</em> (2019-06-19) <a href="https://doi.org/gnj5hm">https://doi.org/gnj5hm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fgene.2019.00597">10.3389/fgene.2019.00597</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31275363">31275363</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6593075">PMC6593075</a></div></div>
</div>
<div id="ref-RRMEOwZ6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>Molecular characterization, computational analysis and expression profiling of &lt;i&gt;Dmrt1&lt;/i&gt; gene in Indian major carp, &lt;i&gt;Labeo rohita&lt;/i&gt; (Hamilton 1822)</strong> <div class="csl-block">L Sahoo, S Sahoo, M Mohanty, M Sankar, S Dixit, P Das, KD Rasal, MA Rather, JK Sundaray</div> <em>Animal Biotechnology</em> (2021-08-02) <a href="https://doi.org/gnj5hj">https://doi.org/gnj5hj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1080/10495398.2019.1707683">10.1080/10495398.2019.1707683</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31880491">31880491</a></div></div>
</div>
<div id="ref-HUT5rWB4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>Revealing liver specific microRNAs linked with carbohydrate metabolism of farmed carp, Labeo rohita (Hamilton, 1822)</strong> <div class="csl-block">Kiran D Rasal, Mir Asif Iquebal, Amrendra Pandey, Parmeswari Behera, Sarika Jaiswal, Manohar Vasam, Sangita Dixit, Mustafa Raza, Lakshman Sahoo, Samiran Nandi, … Jitendra Kumar Sundaray</div> <em>Genomics</em> (2020-01) <a href="https://doi.org/gnj5hh">https://doi.org/gnj5hh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ygeno.2019.07.010">10.1016/j.ygeno.2019.07.010</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31325488">31325488</a></div></div>
</div>
<div id="ref-hpPKbDbP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>De novo Assembly and Genome-Wide SNP Discovery in Rohu Carp, Labeo rohita</strong> <div class="csl-block">Paramananda Das, Lakshman Sahoo, Sofia P Das, Amrita Bit, Chaitanya G Joshi, Basdeo Kushwaha, Dinesh Kumar, Tejas M Shah, Ankit T Hinsu, Namrata Patel, … Joykrushna Jena</div> <em>Frontiers in Genetics</em> (2020-04-21) <a href="https://doi.org/gnj5hn">https://doi.org/gnj5hn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fgene.2020.00386">10.3389/fgene.2020.00386</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32373166">32373166</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7186481">PMC7186481</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
