## Results & Discussion 

<!-- 
Results and Discussion: Should include a description of the genomes, coverage 
amount of genome sequenced, details of sequence variants ascertained, 
identifications of deletions, comparison of strains/genomes (if reporting WGS 
of multiple strains/organisms), and relevance of findings to overall 
organism/strain catalog.
-->

### Sequencing & Assembly 

The C-value of *L. rohita* was previously reported as 1.99 pg (~1.95Gb) using Feulgen densitometry [@genome-size] or 1.5Gb using k-mer estimation [@pmcid:PMC7186481]. 
However, the flow cytometry results [@tbl:flowcyto] show a C-value of 0.99 pg (~0.97Gb) with a standard deviation of 0.02 across all measurements. 
The smaller C-value is also closer to the genome estimate produced by GenomeScope (0.97Gb) and the final genome assembly size of 0.95 Gb.


| Specimen Name     | Number of Sample nuclei | Average sample fluorescence | Number of standard nuclei | Average standard fluorescence | Estimated Genome size ^2^ | HAPLOID       |
|-------------------|-------------------------|-----------------------------|---------------------------|-------------------------------|---------------------------|---------------|
| Fish 1 Sample 1   | 16020                   | 27350                       | 2065                      | 69247                         | 2.049857756               | 1.024928878   |
| Fish 2 Sampe 1    | 13082                   | 25929                       | 6570                      | 66671                         | 2.018441451               | 1.0092207255  |
| Fish 2 Sample 2   | 15402                   | 25665                       | 4354                      | 67489                         | 1.973674969               | 0.9868374845  |
| Fish 3 Sample 1   | 15124                   | 25798                       | 4442                      | 68195                         | 1.963364176               | 0.981682088   |
| Fish 3 Sample 2   | 14923                   | 25763                       | 4823                      | 68837                         | 1.942414254               | 0.971207127   |
| Fish 4 Sample 1   | 13320                   | 26346                       | 5913                      | 69665                         | 1.962760927               | 0.9813804635  |
| Fish 4 Sample 2   | 5624                    | 26612                       | 4097                      | 68876                         | 2.005288925               | 1.0026444625  |
| Fish 5 Sample 1   | 6771                    | 25761                       | 3080                      | 68825                         | 1.942602107               | 0.9713010535  |
| Fish 5 Sample 2   | 15926                   | 26369                       | 3352                      | 68832                         | 1.988248344               | 0.994124172   |
| Standard only ^1^ | 3                       | 25258                       | 3311                      | 64331                         | NA                        |               |
|                   |                         |                             |                           | Average                       | 1.982961434               | 0.9914807172  |
|                   |                         |                             |                           | Standard Deviation            | 0.03607582                | 0.01803790999 |

Table: Flow cytometry results for 5 *L. rohita* blood samples, measured twice. 
1) Trout erthrocyte nuclei: Genome size = 5.19pg.
2) Genome estimate calculated as (average sample fluorescence/ average standard fluorescence * standard genome size ) in picograms DIPLOID (i.e., 2C).
{#tbl:flowcyto}

A total of 130.5 Gb of Nanopore long reads from 44.7 million read, and 261 Gb of Illumina short reads from 870 million pairs were produced, along with 382 million pairs (114 Gb) for the Hi-C library.
The initial *de novo* assembly consisted of 4999 contigs with an N50 of 1.28 Mb. 
After the Bionano and HiC data was incorporated, the total number of sequences dropped to 2899 and the N50 increased to 29.9 Mb.
The final assembly consisted of 25 chromosome length scaffolds and 2844 unplaced scaffolds, ranging in size from 1,479bp to 7.18 Mb.
[@tbl:assembly] contains a list of assembly statistics for each step.
The final genome size is 97.9% of the estimated genome size. 
BUSCO analysis show the genome completely contains 98.1% of the 3640 genes in the actinopterygii_odb10 database with a 36 (1%) genes duplicated, 23 (0.6%) fragmented, and 46 (1.3%) missing.

| n    | n:500 | L50 | min  | N75      | N50      | N25      | E-size   | max      | sum      | name    |
|------|-------|-----|------|----------|----------|----------|----------|----------|----------|---------|
| 4999 | 4999  | 202 | 1348 | 514919   | 1281850  | 2395030  | 1727184  | 7832582  | 9.43E+08 | wtdbg2  |
| 3709 | 3706  | 15  | 1479 | 1.13E+07 | 2.65E+07 | 3.08E+07 | 2.20E+07 | 3.79E+07 | 9.46E+08 | bionano |
| 2899 | 2896  | 14  | 1479 | 2.64E+07 | 2.99E+07 | 3.43E+07 | 2.69E+07 | 4.45E+07 | 9.46E+08 | hic     |
| 2872 | 2869  | 13  | 1479 | 2.88E+07 | 3.25E+07 | 3.61E+07 | 3.00E+07 | 4.53E+07 | 9.46E+08 | ragtag  |
Table: Assembly statistics
{#tbl:assembly}


 
| 238509 | CDS              |
| 239799 | exon             |
| 9824   | five_prime_UTR   |
| 24385  | gene             |
| 24385  | mRNA             |
| 1509   | three_prime_UTR  |
 

```
==> 99-final/Rohu.busco.genome.txt <== 
# BUSCO version is: 5.2.2  
# The lineage dataset is: actinopterygii_odb10 (Creation date: 2020-08-05, number of genomes: 3640, number of BUSCOs: 26) 
# Summarized benchmarking in BUSCO notation for file /work/maa146/rohu-genome/annotations/0-ref/Rohu.genome.fa 
# BUSCO was run in mode: genome 
# Gene predictor used: metaeuk 
  
        ***** Results: ***** 
  
        C:98.1%[S:97.1%,D:1.0%],F:0.6%,M:1.3%,n:3640        
        3571    Complete BUSCOs (C)                         
        3535    Complete and single-copy BUSCOs (S)         
        36      Complete and duplicated BUSCOs (D)          
        23      Fragmented BUSCOs (F)                       
        46      Missing BUSCOs (M)                          
        3640    Total BUSCO groups searched                 
  
```

```
==> 99-final/Rohu.busco.transcript.txt <== 
# BUSCO version is: 5.2.2  
# The lineage dataset is: actinopterygii_odb10 (Creation date: 2020-08-05, number of genomes: 3640, number of BUSCOs: 26) 
# Summarized benchmarking in BUSCO notation for file /work/maa146/rohu-genome/annotations/10-rename/Rohu.transcripts.fa 
# BUSCO was run in mode: transcriptome 
  
        ***** Results: ***** 
  
        C:80.1%[S:78.2%,D:1.9%],F:5.5%,M:14.4%,n:3640       
        2915    Complete BUSCOs (C)                         
        2845    Complete and single-copy BUSCOs (S)         
        70      Complete and duplicated BUSCOs (D)          
        200     Fragmented BUSCOs (F)                       
        525     Missing BUSCOs (M)                          
        3640    Total BUSCO groups searched                 

```

```
==> 99-final/Rohu.busco.unfiltered_transcripts.txt <== 
# BUSCO version is: 5.2.2  
# The lineage dataset is: actinopterygii_odb10 (Creation date: 2020-08-05, number of genomes: 3640, number of BUSCOs: 26) 
# Summarized benchmarking in BUSCO notation for file /work/maa146/rohu-genome/annotations/5-maker/Rohu/all.transcripts.fasta 
# BUSCO was run in mode: transcriptome 
  
        ***** Results: ***** 
  
        C:83.4%[S:81.4%,D:2.0%],F:6.2%,M:10.4%,n:3640       
        3036    Complete BUSCOs (C)                         
        2962    Complete and single-copy BUSCOs (S)         
        74      Complete and duplicated BUSCOs (D)          
        225     Fragmented BUSCOs (F)                       
        379     Missing BUSCOs (M)                          
        3640    Total BUSCO groups searched                 
  
```


```
================================================== 
file name: Rohu.genome.fa            
sequences:          2872 
total length: 1128029156 bp  (945637473 bp excl N/X-runs) 
GC level:         36.05 % 
bases masked:  465289941 bp ( 41.25 %) 
================================================== 
               number of      length   percentage 
               elements*    occupied  of sequence 
-------------------------------------------------- 
Retroelements       123482     53424977 bp    4.74 % 
   SINEs:                0            0 bp    0.00 % 
   Penelope              0            0 bp    0.00 % 
   LINEs:            29838     13842585 bp    1.23 % 
    CRE/SLACS            0            0 bp    0.00 % 
     L2/CR1/Rex      15537      8265099 bp    0.73 % 
     R1/LOA/Jockey     981       643697 bp    0.06 % 
     R2/R4/NeSL        936       732043 bp    0.06 % 
     RTE/Bov-B         271       181676 bp    0.02 % 
     L1/CIN4          7978      1801119 bp    0.16 % 
   LTR elements:     93644     39582392 bp    3.51 % 
     BEL/Pao          1287      1138420 bp    0.10 % 
     Ty1/Copia         221       110739 bp    0.01 % 
     Gypsy/DIRS1     32714     19073624 bp    1.69 % 
       Retroviral     2443      1815904 bp    0.16 % 
  
DNA transposons      79815     22485920 bp    1.99 % 
   hobo-Activator     2242       639067 bp    0.06 % 
   Tc1-IS630-Pogo    49209     16661924 bp    1.48 % 
   En-Spm                0            0 bp    0.00 % 
   MuDR-IS905            0            0 bp    0.00 % 
   PiggyBac              0            0 bp    0.00 % 
   Tourist/Harbinger  1794       259934 bp    0.02 % 
   Other (Mirage,      900       209002 bp    0.02 % 
    P-element, Transib) 
  
Rolling-circles        196        71055 bp    0.01 % 
  
Unclassified:       1774260    361869982 bp   32.08 % 
  
Total interspersed repeats:   437780879 bp   38.81 % 
  
  
Small RNA:               0            0 bp    0.00 % 
  
Satellites:              1          267 bp    0.00 % 
Simple repeats:     504151     23839806 bp    2.11 % 
Low complexity:      58354      3597934 bp    0.32 % 
================================================== 
  
* most repeats fragmented by insertions or deletions 
  have been counted as one element 
                                                       
  
RepeatMasker version 4.1.1 , default mode 
                                         
run with rmblastn version 2.10.0+ 
The query was compared to classified sequences in "Rohu-families.fa" 
 
```
