## Results & Discussion 

<!-- 
Results and Discussion: Should include a description of the genomes, coverage 
amount of genome sequenced, details of sequence variants ascertained, 
identifications of deletions, comparison of strains/genomes (if reporting WGS 
of multiple strains/organisms), and relevance of findings to overall 
organism/strain catalog.
-->

### Sequencing & Assembly 

The C-value of *L. rohita* was previously reported as 1.99 pg (~1.95Gb) using
Feulgen densitometry [@genome-size] or 1.5Gb using k-mer estimation
[@pmcid:PMC7186481]. However, the flow cytometry results (Table [@tbl:flowcyto])
show a C-value of 0.99 pg (~0.97Gb) with a standard deviation of 0.02 across all
measurements. The smaller C-value is also closer to the genome estimate produced
by GenomeScope (0.97Gb) and the final genome assembly size of 0.95 Gb.


| Specimen Name     | Number of Sample nuclei | Average sample fluorescence | Number of standard nuclei | Average standard fluorescence | Estimated Genome size ^2^ | HAPLOID       |
|-------------------|-------------------------|-----------------------------|---------------------------|-------------------------------|---------------------------|---------------|
| Fish 1 Sample 1   | 16020                   | 27350                       | 2065                      | 69247                         | 2.049857756               | 1.024928878   |
| Fish 2 Sample 1   | 13082                   | 25929                       | 6570                      | 66671                         | 2.018441451               | 1.0092207255  |
| Fish 2 Sample 2   | 15402                   | 25665                       | 4354                      | 67489                         | 1.973674969               | 0.9868374845  |
| Fish 3 Sample 1   | 15124                   | 25798                       | 4442                      | 68195                         | 1.963364176               | 0.981682088   |
| Fish 3 Sample 2   | 14923                   | 25763                       | 4823                      | 68837                         | 1.942414254               | 0.971207127   |
| Fish 4 Sample 1   | 13320                   | 26346                       | 5913                      | 69665                         | 1.962760927               | 0.9813804635  |
| Fish 4 Sample 2   | 5624                    | 26612                       | 4097                      | 68876                         | 2.005288925               | 1.0026444625  |
| Fish 5 Sample 1   | 6771                    | 25761                       | 3080                      | 68825                         | 1.942602107               | 0.9713010535  |
| Fish 5 Sample 2   | 15926                   | 26369                       | 3352                      | 68832                         | 1.988248344               | 0.994124172   |
| Standard only ^1^ | 3                       | 25258                       | 3311                      | 64331                         | NA                        |               |
|                   |                         |                             |                           | Average                       | 1.982961434               | 0.9914807172  |
|                   |                         |                             |                           | Standard Deviation            | 0.03607582                | 0.01803790999 |

Table: Flow cytometry results for 5 *L. rohita* blood samples, measured twice. 
1) Trout erythrocyte nuclei: Genome size = 5.19pg.
2) Genome estimate calculated as (average sample fluorescence/ average standard fluorescence * standard genome size) in picograms DIPLOID (i.e., 2C).
{#tbl:flowcyto}

A total of 130.5 Gb of Nanopore long reads from 44.7 million read, and 261 Gb of
Illumina short reads from 870 million pairs were produced, along with 382
million pairs (114 Gb) for the Hi-C library. The initial *de novo* assembly
consisted of 4999 contigs with an N50 of 1.28 Mb. After the Bionano and HiC data
was incorporated, the total number of sequences dropped to 2899 and the N50
increased to 29.9 Mb. The final assembly consists of 25 chromosome length
scaffolds and 2844 unplaced scaffolds, ranging in size from 1,479bp to 7.18 Mb.
The chromosome scaffolds consist of one to eight sequences with all but three
made from three or fewer sequences. Table [@tbl:assembly] contains a common
assembly statistics for each step. The final genome size is 97.9% of the
estimated genome size. RepeatMasker masked 41.25% of the genome. Unfortunately,
given the ad hoc nature of the repeat database created by RepeatModeler2, most (92.5%)
of the repeats found were unclassified. The annotation pipeline produced 51,079
primary transcripts, 31,274 surviving the AED, gap, and overlapping filters.
BUSCO analysis show the genome completely contains 98.1% of the 3640 orthologs
in the actinopterygii_odb10 database with 37 (1%) duplicated; while, the
filtered transcriptome contains 84.5% of the total orthologs complete with 74
duplicated. A complete comparison of the BUSCO analyses can be found in Table
[@tbl:busco].

| n    | n:500 | L50 | min  | N75      | N50      | N25      | E-size   | max      | sum      | name    |
|------|-------|-----|------|----------|----------|----------|----------|----------|----------|---------|
| 4999 | 4999  | 202 | 1348 | 514919   | 1281850  | 2395030  | 1727184  | 7832582  | 9.43E+08 | wtdbg2  |
| 3709 | 3706  | 15  | 1479 | 1.13E+07 | 2.65E+07 | 3.08E+07 | 2.20E+07 | 3.79E+07 | 9.46E+08 | bionano |
| 2899 | 2896  | 14  | 1479 | 2.64E+07 | 2.99E+07 | 3.43E+07 | 2.69E+07 | 4.45E+07 | 9.46E+08 | hic     |
| 2872 | 2869  | 13  | 1479 | 2.88E+07 | 3.25E+07 | 3.61E+07 | 3.00E+07 | 4.53E+07 | 9.46E+08 | ragtag  |

Table: Assembly statistics for each stage of the assembly
{#tbl:assembly}

| Type                                | Genome | Unfiltered Transcriptome | Filtered Transcriptome |
|-------------------------------------|--------|--------------------------|------------------------|
| Complete BUSCOs (C)                 | 3571   | 3139                     | 3078                   |
| Complete and single-copy BUSCOs (S) | 3534   | 3064                     | 3001                   |
| Complete and duplicated BUSCOs (D)  | 37     | 75                       | 74                     |
| Fragmented BUSCOs (F)               | 23     | 192                      | 170                    |
| Missing BUSCOs (M)                  | 46     | 309                      | 392                    |
| Total BUSCO groups searched         | 3640   | 3640                     | 3640                   |

Table: BUSCO analysis for the genome and transcriptome, before and after AED filtering.
{#tbl:busco}


### Comparative genomics

#### Labeo Rohita

Our assembly (IGBB) was compared with the publicly available rohu assemblies
(CIFA and HRRL) using several metrics. Both the scaffold N50 and maximum length
of our assembly are 30Mb higher than either of the other two
[@tbl:assembly-stats]. The length distributions ([@fig:len-distro]) show a
similar separation. The IGBB assembly has a hight overall completeness, along
with fewer duplicate single-copy orthologs, than either the CIFA or HRRL
assemblies ([@fig:busco-rohu]). Lastly, all assemblies were pairwise aligned and
plotted ([@fig:dotplot-rohu]). All assemblies are largely similar with the CIFA
and HRRL assemblies showing a near perfect alignment with each other. When
compared with the IGBB assembly, both assemblies show a few errors, specifically
in Chr09 and Chr19. Due to the size difference between the assemblies, the
inverse comparisons where not informative.

| Name         | CIFA    | HRRL    | IGBB    |
|--------------|---------|---------|---------|
| Scaffolds    | 13623   | 13660   | 2869    |
| L50          | 182     | 182     | 13      |
| N75          | 774698  | 774698  | 28.79e6 |
| N50          | 2006958 | 2006958 | 32.5e6  |
| N25          | 4279019 | 4279019 | 36.06e6 |
| E-size       | 2905260 | 2905216 | 30.03e6 |
| Max Scaffold | 15.15e6 | 15.15e6 | 45.29e6 |
| Total Length | 1.427e9 | 1.427e9 | 945.5e6 |

Table: Assembly stats for rohu genomes
{#tbl:assembly-stats}

![
Length distribution of scaffolds for each rohu assembly.
Each bar represents the number of bases contained in scaffolds of a particular length.
](../comp-gen/1-stats/length.rohu.hires.png "Length distro")[#fig:len-distro]

![
BUSCO score for each rohu genome
](../comp-gen/1-busco/rohu/busco.summary.png "Rohu BUSCOs")[#fig:busco-rohu]

![
Dot-plots comparing sequence similarity between rohu genomes
](../comp-gen/2-dotplot/rohu.png)[#fig:dotplot-rohu]

#### Cypriniformes

The IGBB assembly was compared with the available annotated Cypriniformes
assemblies with a scaffold level or higher from NCBI. The IGBB assembly had the
most complete, single-copy BUSCOs of any Cypriniformes ([@fig:busco-carps]).

Dot-plots were used to compare the chromosome level assembles
([@fig:dotplot-carps]). The assemblies for *Danio rerio* and *Triplophysa
tibetana* were removed from the dot plot grid since very few of the alignments
passed the graphing threshold. *Carassius auratus* and *Cyprinus carpio* are
both allotetraploids [@doi:10.1016/j.cub.2020.05.073],
[@doi:10.1038/s41467-019-12644-1] and thus show a good deal of duplication in
the dot-plots. The three diploids included show similar chromosome structure
with some rearrangements.

Lastly, the annotations for the Cypriniformes were compared using OrthoFinder.
Of the 31,274 genes annotated, 29,904 (95.6%) were placed into 18,740
orthogroups, 63.5% of the total orthogroups found. Table [@tbl:orthofinder]
contains the summary statistics for all species used in the OrthoFinder
analysis.


![
BUSCO score for each rohu genome
Sorted by complete single-copy BUSCO count. Sections including more than 150 BUSCOs are labeled. 
](../comp-gen/1-busco/cypriniformes/busco.summary.png "Rohu BUSCOs")[#fig:busco-carps]

![
Dot-plots comparing sequence similarity between rohu genomes
](../comp-gen/2-dotplot/cypriniformes.rm-uninformative.png)[#fig:dotplot-carps]


|                              | Number of Genes | Genes in Orthogroups | Unassigned Genes | Orthogroups Containing Species | Species-specific Orthogroups | Genes in Species-specific Orthogroups |
|------------------------------|-----------------|----------------------|------------------|--------------------------------|------------------------------|---------------------------------------|
| Anabarilius grahami          | 23906           | 21942 (91.8%)        | 1964 (8.2%)      | 15217 (51.5%)                  | 180                          | 888 (3.7%)                            |
| Carassius auratus            | 96703           | 93988 (97.2%)        | 2715 (2.8%)      | 21896 (74.2%)                  | 472                          | 1775 (1.8%)                           |
| Cyprinus carpio              | 80686           | 78684 (97.5%)        | 2002 (2.5%)      | 21584 (73.1%)                  | 287                          | 964 (1.2%)                            |
| Danio rerio                  | 52829           | 51951 (98.3%)        | 878  (1.7%)      | 20671 (70.0%)                  | 392                          | 2184 (4.1%)                           |
| Danionella translucida       | 35381           | 32943 (93.1%)        | 2438 (6.9%)      | 19073 (64.6%)                  | 540                          | 1941 (5.5%)                           |
| Labeo rohita                 | 31274           | 29904 (95.6%)        | 1370 (4.4%)      | 18740 (63.5%)                  | 161                          | 1581 (5.1%)                           |
| Onychostoma macrolepis       | 24754           | 24483 (98.9%)        | 271  (1.1%)      | 19276 (65.3%)                  | 137                          | 603 (2.4%)                            |
| Pimephales promelas          | 47578           | 45412 (95.4%)        | 2166 (4.6%)      | 19884 (67.4%)                  | 506                          | 1826 (3.8%)                           |
| Puntigrus tetrazona          | 48681           | 48094 (98.8%)        | 587  (1.2%)      | 20582 (69.7%)                  | 129                          | 517 (1.1%)                            |
| Sinocyclocheilus anshuiensis | 68474           | 66456 (97.1%)        | 2018 (2.9%)      | 21485 (72.8%)                  | 114                          | 344 (0.5%)                            |
| Sinocyclocheilus grahami     | 67410           | 63316 (93.9%)        | 4094 (6.1%)      | 22326 (75.6%)                  | 338                          | 793 (1.2%)                            |
| Sinocyclocheilus rhinocerous | 68562           | 65831 (96.0%)        | 2731 (4.0%)      | 21884 (74.1%)                  | 172                          | 414 (0.6%)                            |
| Triplophysa tibetana         | 24310           | 23279 (95.8%)        | 1031 (4.2%)      | 18734 (63.5%)                  | 125                          | 480 (2.0%)                            |

Table: Summary statistics per species from OrthoFinder.
{#tbl:orthofinder}

_SNP discovery and population similarities among rohu fisheries_

_Labeo rohita_ is an important aquaculture fish in many areas of south Asia [I
donâ€™t know how to put in this reference FAO (2019) Food and Agricultural
Organization. Yearbook of Fishery and Aquaculture Statistics 2017].

The high fecundity and relative ease of cultivation has made rohu successful in
breeding programs, leading to the generation of a fast-growing strain called
Jayanti Historically, rohu breeding and research has relied on low-throughput
information, such as cloning individual genes of interest (reviewed in
[@doi:10.1111/raq.12444]) or microsatellite-based selective breeding
[@doi:10.1007/s10499-016-0104-z]; however, recent years have seen increased
interest in applying high-throughput methods to rohu breeding, such as SNP
panels [@doi:10.1186/1471-2164-15-541] [@doi:10.1016/J.GENREP.2016.12.004],
genotyping-by-sequencing [@doi:10.3389/fgene.2019.00597], transcriptomics
[@doi:10.1186/1471-2164-15-541] [@doi:10.1080/10495398.2019.1707683], and
degradomics [@doi:10.1016/j.ygeno.2019.07.010]. Recently, a draft genome was
published for L. rohita along with pooled resequencing from three individuals in
10 populations [@doi:10.3389/fgene.2020.00386]; however, genetic diversity and
divergence among the populations was not reported.













