#+TITLE: Population Structure
#+PROPERTY:  header-args :var DIR=(file-name-directory buffer-file-name)

* PCA

#+header: :results output graphics file :file fish.pca.png
#+begin_src R :width 6 :height 6 :units in :res 300
#+begin_src R
library(tidyverse)
library(ggrepel)
library(SNPRelate)



#load in VCF file
vcf.fn <- "fish.pass2.90pct.all.recode.vcf.gz"

#Reformat
snpgdsVCF2GDS(vcf.fn, "fish.pass2.90pct.tmp.gds", method="biallelic.only")
snpgdsSummary("fish.pass2.90pct.tmp.gds")

#Open the GDS file
genofile <- snpgdsOpen("fish.pass2.90pct.tmp.gds")

#Get create vectors from genofile
samplenames <- read.gdsn(index.gdsn(genofile, path = "sample.id" ))

############# Turning the sample.id list into a population code list.
pop_code <- read.csv("Rohu_sampling_data.csv", header =T)
pop_code <- pop_code[,c(2,4,5,7)]
names(pop_code)[1] <- "sample.id"

#Remove outgroup
sample.id <- samplenames[!(samplenames %in% c("SRR12102514"))]

                                        #Principal Component Analysis(PCA)
pca <- snpgdsPCA(genofile,
                 sample.id=sample.id,
                 num.thread=12,
                 autosome.only=F)
pc.percent <- pca$varprop*100

tab <- data.frame(sample.id = pca$sample.id,
                  EV1 = pca$eigenvect[,1],    # the first eigenvector
                  EV2 = pca$eigenvect[,2],    # the second eigenvector
                  stringsAsFactors = FALSE)

tab$sample.id[tab$sample.id == "BFA0224"] = "BFA0229"
tab$sample.id[tab$sample.id == "RBFA0431"] = "BFA0431"

PCAtab <- merge(tab,pop_code, by="sample.id", all.x=T)

ggplot(PCAtab, aes(x=EV2, y=EV1, color=Population)) +
  geom_point(alpha=1) +
  scale_color_brewer(type='qual', palette="Set1") +
  xlab(sprintf("EV2 (%.2f%%)", pc.percent[2])) +
  ylab(sprintf("EV1 (%.2f%%)", pc.percent[1])) +
  theme_minimal()


#+end_src

#+RESULTS:
[[file:fish.pca.png]]

* LEA
1) convert vcf to ped file
  #+begin_src sh :tangle 1-convert.sh
ROOT=$(git rev-parse --show-toplevel)

ml singularity
vcftools () {
    singularity exec -B$ROOT \
        /apps/singularity-3/vcftools/vcftools-v0.1.16-1-deb_cv1.sif \
        vcftools $@
}


awk '{print $1, NR}' OFS="\t" FS="\t" \
    $ROOT/assembly/6-ragtag/high-conf-scaff/ragtag.scaffolds.fasta.fai \
    > $DIR/chr-map.all


# Remove lines with missing alt
zcat $DIR/fish.pass2.90pct.all.recode.vcf.gz |
    awk '/^#/ || $5 != "."' > $DIR/fish.vcf

vcftools --vcf $DIR/fish.vcf \
    --remove-indv SRR12102514 \
    --chrom-map $DIR/chr-map.all \
    --remove-indels \
    --plink \
    --out $DIR/fish

  #+end_src

2) Run LEA
   #+NAME: run-lea
   #+begin_src R :tangle 2-run-lea.R
library(LEA)


ped2geno("fish.ped", "fish.geno")
#vcf2geno("fish.vcf", "fish.geno")
snmf("fish.geno",
     K = 1:10,
     entropy = T,
     repetitions = 10,
     project = "new",
     ploidy=2,
     CPU = 48)

   #+end_src

3) Cross-Entropy
   #+name: graph-entropy
   #+header: :var project.dir="fish.snmfProject"
   #+begin_src R
library(tidyverse)
library(LEA)
project.dir<-"fish.snmfProject"
project.dir <- scan(text=project.dir, what="character", allowEscapes = T)
#Load snmf project
project = load.snmfProject(project.dir)

# Get best Ks by finding minimum min, mean, and max cross entropy
summary.info = summary(project)$crossEntropy

                                        # Plot the mean cross entropy.
# Get the mean from summary_info and convert the rownames to column "K".
# Remove "K = " from the K column.
# Rename the "summary_info[2, ]" column to "mean".
# Pass this information to ggplot.
# Add a horizontal red line at the minimum cross entropy.
# Plot the cross entropy by K as a scatterplot.
# Set the x-axis labels from 1-10.
# Change the x-axis and y-axis labels and the title.
# Remove the legend.
as.data.frame(t(summary.info)) %>%
  tibble::rownames_to_column("K") %>%
  mutate(K = factor(as.numeric(substring(K, 4)))) %>%
  ggplot() +
  geom_hline(aes(yintercept = min(summary.info[2,]), color = "red")) +
  geom_pointrange(aes(x = K, ymin=min, y = mean, ymax=max)) +
  labs(title = "Cross-entropy versus K", x = "Number of ancestral populations (K)", y = "Cross-entropy") +
  theme_minimal() +
  theme(legend.position = "none",
        panel.grid.major.x=element_blank(),
        panel.grid.minor.x=element_blank(),
        plot.background = element_rect(fill="white"))

ggsave("fish.entropy.png", scale=2, width=4, height=3.5)

  #+end_src

  #+RESULTS: graph-entropy
  [[file:fish.entropy.png]]

  [[./fish.entropy.png]]

   #+name: selected-runs
   #+header: :var project.dir="fish.snmfProject"
   #+begin_src R :colnames yes :rownames yes
library(LEA)

project.dir <- scan(text=project.dir, what="character", allowEscapes = T)
project = load.snmfProject(project.dir)


## Get relevent data for each run
summary = data.frame(
  'K' = sapply(project@runs, slot, "K"),
  'run' = sapply(project@runs, slot, "run"),
  'crossEntropy' = sapply(project@runs, slot, "crossEntropy"),
  'file' = sapply(project@runs, slot, "Q.output.file"))

## Find the top 5 with lowest cross entropy
index = head(order(summary$crossEntropy), n=5)

summary[index,]
   #+end_src

   #+RESULTS: selected-runs
   |    | K | run |       crossEntropy | file         |
   |----+---+-----+--------------------+--------------|
   | 51 | 1 |   6 | 0.0783959533321844 | fish_r6.1.Q  |
   | 91 | 1 |  10 | 0.0784233170853525 | fish_r10.1.Q |
   | 71 | 1 |   8 | 0.0784372243261066 | fish_r8.1.Q  |
   | 52 | 2 |   6 | 0.0785278984525492 | fish_r6.2.Q  |
   | 92 | 2 |  10 | 0.0785511115852912 | fish_r10.2.Q |

4) Membership

     #+header: :var prefix="fish"
     #+header: :results output graphics file :file fish.structure.png
     #+begin_src R :width 4800 :height 1800
library(data.table)
library(tidyverse)
library(hues)
library(LEA)

prefix <- scan(text=prefix, what="character", allowEscapes = T)
project = load.snmfProject(paste0(prefix, ".snmfProject"))

sample.names <- fread(paste0(prefix, ".ped"),
                      select = 1,
                      header=F,
                      stringsAsFactors=F,
                      col.names=c("sample"))


meta.data <- read.delim("../raw/attributes.tsv")


r = data.frame("K"=2, "Run"=6)

  raw.data <- as.data.frame(Q(project, K=r$K, run=r$Run))

  data.order <- dist(raw.data) %>%
    hclust(method="ave") %>%
    as.dendrogram %>%
    order.dendrogram

  data <- cbind(sample=sample.names$sample, raw.data)

  ordered.data <- data[data.order, ] %>%
    merge(meta.data, by.x="sample", by.y="sample_name", all.x=T) %>%
    select("sample", starts_with('V'), "location"="ecotype") %>%
    filter(!is.na(location)) %>%
    mutate(sample=fct_reorder(factor(sample), V1)) %>%
    gather(-c("sample", "location"), key="cluster", value="percent")

plot<-  ggplot(ordered.data, aes(y = percent, x =sample, fill=cluster)) +
    geom_col(width = 1) +
    labs(title = paste0("Membership in ", r$K, " Inferred Populations"),
         y = "Membership",
         x = "Sample",
         fill = "Population") +
    scale_y_continuous(labels=scales::percent, expand = c(0,0)) +
    scale_x_discrete(expand = c(0,0)) +
    theme_minimal() +
    theme( plot.background = element_rect(fill="white"),
          axis.text.x=element_blank(),
          legend.position = 'bottom') +
    guides(fill="none") +
    facet_grid(cols=vars(location),scales="free_x", space = "free")

  plot + scale_fill_manual(values = c("#61acd0","#3d7287"))


ggsave("fish.structure.png", scale=2, width=8, height=2.5)

     #+end_src

     #+RESULTS:
     [[file:fish.structure.png]]

     [[./fish.structure.png]]

* Figure 3
   Combine diversity/divergence map, entropy, and structure into one figure

   #+begin_src sh
convert \( \( \( -resize x1800 Figure\ Fish\ Map.jpg \) \
              \( -background transparent -fill grey -pointsize 150 \
                 -size 1800x1800 -gravity center label:A \
                 -trim +repage -bordercolor none -border 10x10 \) \
              -gravity northwest -composite \) \
           \( \( -resize x1800 fish.pca.png \) \
              \( -fill grey -pointsize 150 -size 1800x1800 -gravity center \
                    label:C -trim +repage -bordercolor none -border 10x10 \) \
              -gravity northwest -composite \) \
           +append -resize 4000x \) \
        \( \( \( -resize x1000 fish.structure.png \) \
              \( -fill grey -pointsize 150 -size 1800x1800 -gravity center \
                 label:B -trim +repage -bordercolor none -border 10x10 \) \
              -gravity northwest -composite \) \
           \( \( -resize x1000 fish.entropy.png \) \
              \( -fill grey -pointsize 150 -size 1800x1800 -gravity center \
                 label:D -trim +repage -bordercolor none -border 10x10 \) \
              -gravity northwest -composite \) \
        +append -resize 4000x \) \
    -append fig3.png

   #+end_src

   [[./fig3.png]]
   #+RESULTS:

* Supplemental Figure Manhattan
   #+begin_src sh
convert \( \( -resize x1800 pixyFst.jpg \) \
           \( -background transparent -fill grey -pointsize 150 \
              -size 1800x1800 -gravity center label:A \
              -trim +repage -bordercolor none -border 10x10 \) \
           -gravity northwest -composite \) \
        \( \( -resize x1800 pixyPi.jpg \) \
           \( -background transparent -fill grey -pointsize 150 \
              -size 1800x1800 -gravity center label:B \
              -trim +repage -bordercolor none -border 10x10 \) \
           -gravity northwest -composite \) \
        \( \( -resize x1800 pixyDxy.jpg \) \
           \( -background transparent -fill grey -pointsize 150 \
              -size 1800x1800 -gravity center label:C \
              -trim +repage -bordercolor none -border 10x10 \) \
           -gravity northwest -composite \) \
    -append supp.fig.manhattan.png

   #+end_src

[[./supp.fig.manhattan.png]]

* Supplemental Figure Summary
#+name: pi
| Chr        | type |   Halda |  Jamuna |   Padma |
| Overall    | sum  | 0.00063 | 0.00065 | 0.00064 |
| Chromosome | sum  | 0.00057 | 0.00058 | 0.00058 |
| Chr01      | chr  | 0.00056 | 0.00059 | 0.00058 |
| Chr02      | chr  | 0.00057 | 0.00057 | 0.00057 |
| Chr03      | chr  | 0.00076 | 0.00080 | 0.00078 |
| Chr04      | chr  | 0.00073 | 0.00077 | 0.00076 |
| Chr05      | chr  | 0.00053 | 0.00053 | 0.00053 |
| Chr06      | chr  | 0.00048 | 0.00050 | 0.00049 |
| Chr07      | chr  | 0.00052 | 0.00053 | 0.00053 |
| Chr08      | chr  | 0.00060 | 0.00063 | 0.00061 |
| Chr09      | chr  | 0.00054 | 0.00057 | 0.00056 |
| Chr10      | chr  | 0.00054 | 0.00056 | 0.00056 |
| Chr11      | chr  | 0.00054 | 0.00055 | 0.00054 |
| Chr12      | chr  | 0.00050 | 0.00052 | 0.00052 |
| Chr13      | chr  | 0.00051 | 0.00052 | 0.00051 |
| Chr14      | chr  | 0.00056 | 0.00058 | 0.00057 |
| Chr15      | chr  | 0.00058 | 0.00060 | 0.00058 |
| Chr16      | chr  | 0.00053 | 0.00054 | 0.00053 |
| Chr17      | chr  | 0.00053 | 0.00055 | 0.00054 |
| Chr18      | chr  | 0.00053 | 0.00054 | 0.00053 |
| Chr19      | chr  | 0.00062 | 0.00063 | 0.00062 |
| Chr20      | chr  | 0.00049 | 0.00052 | 0.00051 |
| Chr21      | chr  | 0.00051 | 0.00052 | 0.00052 |
| Chr22      | chr  | 0.00085 | 0.00086 | 0.00085 |
| Chr23      | chr  | 0.00053 | 0.00055 | 0.00054 |
| Chr24      | chr  | 0.00054 | 0.00055 | 0.00055 |
| Chr25      | chr  | 0.00056 | 0.00057 | 0.00057 |

#+name: fst
| Chr        | type | Jamuna-Halda | Padma-Halda | Padma-Jamuna |
| Overall    | sum  |       0.0108 |      0.0115 |       0.0049 |
| Chr01      | chr  |       0.0110 |      0.0120 |       0.0052 |
| Chr02      | chr  |       0.0116 |      0.0125 |       0.0041 |
| Chr03      | chr  |       0.0116 |      0.0127 |       0.0056 |
| Chr04      | chr  |       0.0116 |      0.0127 |       0.0047 |
| Chr05      | chr  |       0.0107 |      0.0113 |       0.0050 |
| Chr06      | chr  |       0.0113 |      0.0117 |       0.0045 |
| Chr07      | chr  |       0.0098 |      0.0102 |       0.0051 |
| Chr08      | chr  |       0.0105 |      0.0112 |       0.0051 |
| Chr09      | chr  |       0.0119 |      0.0117 |       0.0050 |
| Chr10      | chr  |       0.0115 |      0.0112 |       0.0044 |
| Chr11      | chr  |       0.0101 |      0.0115 |       0.0050 |
| Chr12      | chr  |       0.0106 |      0.0109 |       0.0052 |
| Chr13      | chr  |       0.0112 |      0.0125 |       0.0050 |
| Chr14      | chr  |       0.0102 |      0.0108 |       0.0042 |
| Chr15      | chr  |       0.0102 |      0.0117 |       0.0048 |
| Chr16      | chr  |       0.0095 |      0.0099 |       0.0046 |
| Chr17      | chr  |       0.0107 |      0.0120 |       0.0048 |
| Chr18      | chr  |       0.0116 |      0.0130 |       0.0052 |
| Chr19      | chr  |       0.0105 |      0.0117 |       0.0054 |
| Chr20      | chr  |       0.0102 |      0.0104 |       0.0044 |
| Chr21      | chr  |       0.0098 |      0.0103 |       0.0050 |
| Chr22      | chr  |       0.0131 |      0.0137 |       0.0054 |
| Chr23      | chr  |       0.0099 |      0.0111 |       0.0050 |
| Chr24      | chr  |       0.0096 |      0.0106 |       0.0050 |
| Chr25      | chr  |       0.0107 |      0.0114 |       0.0049 |

#+name: dxy
| Chr        | type | Jamuna-Halda | Padma-Halda | Padma-Jamuna |
| Overall    | sum  |      0.00065 |     0.00064 |      0.00065 |
| Chromosome | sum  |      0.00058 |     0.00058 |      0.00058 |
| Chr01      | chr  |      0.00058 |     0.00058 |      0.00059 |
| Chr02      | chr  |      0.00058 |     0.00058 |      0.00057 |
| Chr03      | chr  |      0.00079 |     0.00079 |      0.00079 |
| Chr04      | chr  |      0.00076 |     0.00076 |      0.00077 |
| Chr05      | chr  |      0.00054 |     0.00054 |      0.00053 |
| Chr06      | chr  |      0.00050 |     0.00049 |      0.00049 |
| Chr07      | chr  |      0.00053 |     0.00053 |      0.00053 |
| Chr08      | chr  |      0.00062 |     0.00061 |      0.00062 |
| Chr09      | chr  |      0.00056 |     0.00055 |      0.00056 |
| Chr10      | chr  |      0.00056 |     0.00056 |      0.00056 |
| Chr11      | chr  |      0.00055 |     0.00055 |      0.00054 |
| Chr12      | chr  |      0.00052 |     0.00051 |      0.00052 |
| Chr13      | chr  |      0.00052 |     0.00052 |      0.00052 |
| Chr14      | chr  |      0.00058 |     0.00057 |      0.00058 |
| Chr15      | chr  |      0.00060 |     0.00059 |      0.00059 |
| Chr16      | chr  |      0.00054 |     0.00054 |      0.00054 |
| Chr17      | chr  |      0.00055 |     0.00054 |      0.00054 |
| Chr18      | chr  |      0.00054 |     0.00054 |      0.00054 |
| Chr19      | chr  |      0.00063 |     0.00063 |      0.00063 |
| Chr20      | chr  |      0.00051 |     0.00050 |      0.00051 |
| Chr21      | chr  |      0.00052 |     0.00052 |      0.00052 |
| Chr22      | chr  |      0.00087 |     0.00086 |      0.00086 |
| Chr23      | chr  |      0.00055 |     0.00055 |      0.00055 |
| Chr24      | chr  |      0.00055 |     0.00055 |      0.00055 |
| Chr25      | chr  |      0.00057 |     0.00057 |      0.00057 |

#+header: :var data=pi :session
#+header: :width 8 :height 4 :res 600 :units in
#+header: :results graphics file :file summary-pi.png
#+begin_src R
library(tidyverse)
library(janitor)


data <- data %>% row_to_names(1) %>%
  gather(-Chr,-type, key="River", value="value") %>%
  mutate(value = as.numeric(value))

ggplot(data, aes(Chr, value, color=River, shape=River)) +
  geom_point() +
  scale_shape_manual(values=1:3) +
  scale_y_continuous(labels=scales::label_number()) +
  ylab(expression(pi)) +
  facet_grid(~type, scales="free_x", space="free_x") +
  ggtitle("Mean within population nucleotide diversity (π)") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle=45, hjust=1, vjust=1),
        axis.title.x =element_blank(),
        strip.text = element_blank(),
        panel.grid.major.x = element_blank())

#+end_src

#+RESULTS:
[[file:summary-pi.png]]

#+header: :var data=dxy :session
#+header: :width 8 :height 4 :res 600 :units in
#+header: :results graphics file :file summary-dxy.png
#+begin_src R
library(tidyverse)
library(janitor)


data <- data %>% row_to_names(1) %>%
  gather(-Chr,-type, key="River", value="value") %>%
  mutate(value = as.numeric(value))

ggplot(data, aes(Chr, value, color=River, shape=River)) +
  geom_point() +
  scale_shape_manual(values=1:3) +
  scale_y_continuous(labels=scales::label_number()) +
  ylab(expression(pi * "xy")) +
  facet_grid(~type, scales="free_x", space="free_x") +
  ggtitle("Mean between population nucleotide divergence (πxy)" ) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle=45, hjust=1, vjust=1),
        axis.title.x =element_blank(),
        strip.text = element_blank(),
        panel.grid.major.x = element_blank())

#+end_src

#+RESULTS:
[[file:summary-dxy.png]]

#+header: :var data=fst :session
#+header: :width 8 :height 4 :res 600 :units in
#+header: :results graphics file :file summary-fst.png
#+begin_src R
library(tidyverse)
library(janitor)


data <- data %>% row_to_names(1) %>%
  gather(-Chr,-type, key="River", value="value") %>%
  mutate(value = as.numeric(value))

ggplot(data, aes(Chr, value, color=River, shape=River)) +
  geom_point() +
  scale_shape_manual(values=1:3) +
  scale_y_continuous(labels=scales::label_number()) +
  ylab("Fst") +
  facet_grid(~type, scales="free_x", space="free_x") +
  ggtitle("Mean Weir and Cockerham's Fst between fish populations") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle=45, hjust=1, vjust=1),
        axis.title.x =element_blank(),
        strip.text = element_blank(),
        panel.grid.major.x = element_blank())

#+end_src

#+RESULTS:
[[file:summary-fst.png]]

   #+begin_src sh
convert \( \( -resize x1800 summary-fst.png \) \
           \( -background transparent -fill grey -pointsize 150 \
              -size 1800x1800 -gravity center label:A \
              -trim +repage -bordercolor none -border 10x10 \) \
           -gravity northwest -composite \) \
        \( \( -resize x1800 summary-pi.png \) \
           \( -background transparent -fill grey -pointsize 150 \
              -size 1800x1800 -gravity center label:B \
              -trim +repage -bordercolor none -border 10x10 \) \
           -gravity northwest -composite \) \
        \( \( -resize x1800 summary-dxy.png \) \
           \( -background transparent -fill grey -pointsize 150 \
              -size 1800x1800 -gravity center label:C \
              -trim +repage -bordercolor none -border 10x10 \) \
           -gravity northwest -composite \) \
    -append supp.fig.summary.png

   #+end_src

[[./supp.fig.summary.png]]
